// @anon-aadhaar/react/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AnonAadhaarProof: () => AnonAadhaarProof,
  AnonAadhaarProvider: () => AnonAadhaarProvider,
  LaunchProveModal: () => LaunchProveModal,
  LogInWithAnonAadhaar: () => LogInWithAnonAadhaar,
  processAadhaarArgs: () => processAadhaarArgs,
  proveAndSerialize: () => proveAndSerialize,
  useAnonAadhaar: () => useAnonAadhaar,
  useProver: () => useProver,
  verifySignature: () => verifySignature
});
module.exports = __toCommonJS(src_exports);

// src/hooks/useAnonAadhaar.ts
var import_react = require("react");
var import_core = require("@anon-aadhaar/core");
function useAnonAadhaar() {
  const val = (0, import_react.useContext)(AnonAadhaarContext);
  return [val.state, val.startReq];
}
var AnonAadhaarContext = (0, import_react.createContext)({
  state: { status: "logged-out" },
  startReq: () => {
  },
  appName: "",
  proverState: import_core.ProverState.Initializing,
  useTestAadhaar: false
});

// src/provider/AnonAadhaarProvider.tsx
var import_react2 = require("react");
var import_core4 = require("@anon-aadhaar/core");
var import_react3 = __toESM(require("react"));

// src/prove.ts
var import_core3 = require("@anon-aadhaar/core");

// src/verifySignature.ts
var import_core2 = require("@anon-aadhaar/core");

// src/util.ts
var import_jsqr = __toESM(require("jsqr"));
var uploadQRpng = (e, setQrStatus) => {
  return new Promise((resolve, reject) => {
    if (e.target.files) {
      try {
        const fileReader = new FileReader();
        fileReader.readAsDataURL(e.target.files[0]);
        fileReader.onload = (e2) => {
          if (e2.target && e2.target.result) {
            try {
              const image = new Image();
              image.onload = () => {
                const canvas = document.createElement("canvas");
                canvas.width = image.width;
                canvas.height = image.height;
                const ctx = canvas.getContext("2d");
                if (!ctx)
                  throw Error("Image cannot be reconstructed");
                ctx.drawImage(image, 0, 0);
                const imageData = ctx.getImageData(
                  0,
                  0,
                  image.width,
                  image.height
                );
                const qrValue = (0, import_jsqr.default)(imageData.data, image.width, image.height);
                if (qrValue != null) {
                  resolve({
                    qrValue: qrValue.data
                  });
                  setQrStatus("QR code scanned, verifying QR Code \u{1F50E}" /* QR_CODE_SCANNED */);
                } else {
                  setQrStatus("QR code invalid \u274C" /* ERROR_PARSING_QR */);
                }
              };
              image.src = e2.target.result.toString();
            } catch (error) {
              setQrStatus("QR code invalid \u274C" /* ERROR_PARSING_QR */);
              console.error(error);
              reject(error);
            }
          }
        };
      } catch {
        setQrStatus(null);
        reject(new Error("No file selected"));
      }
    }
  });
};
function str2ab(str) {
  const buf = new ArrayBuffer(str.length);
  const bufView = new Uint8Array(buf);
  for (let i = 0, strLen = str.length; i < strLen; i++) {
    bufView[i] = str.charCodeAt(i);
  }
  return buf;
}
var createBlobURL = (icon) => {
  const blob = new Blob([icon], { type: "image/svg+xml" });
  return URL.createObjectURL(blob);
};

// src/verifySignature.ts
var import_node_forge = require("node-forge");

// src/publicKeys.ts
var testCertificate = `-----BEGIN CERTIFICATE-----
MIID6jCCAtKgAwIBAgIBATANBgkqhkiG9w0BAQsFADBpMRQwEgYDVQQDEwtleGFt
cGxlLm9yZzELMAkGA1UEBhMCVVMxETAPBgNVBAgTCFZpcmdpbmlhMRMwEQYDVQQH
EwpCbGFja3NidXJnMQ0wCwYDVQQKEwRUZXN0MQ0wCwYDVQQLEwRUZXN0MB4XDTI0
MDMwMTE2MTc0MFoXDTI1MDMwMTE2MTc0MFowaTEUMBIGA1UEAxMLZXhhbXBsZS5v
cmcxCzAJBgNVBAYTAlVTMREwDwYDVQQIEwhWaXJnaW5pYTETMBEGA1UEBxMKQmxh
Y2tzYnVyZzENMAsGA1UEChMEVGVzdDENMAsGA1UECxMEVGVzdDCCASIwDQYJKoZI
hvcNAQEBBQADggEPADCCAQoCggEBAJXoH3UGWVzCYq3c/d012AQFWyrfRtxxnFa1
ju4ZmZXvoaMlRxBiCseAHpdvROO+RU2w8ZDj99TjWYlyEXNE3lL894JvhJSIqVmn
s9Iett0DRRZi6og+7u/eiJoUmbmkf5UExfCWwmK5bSPRl1AzLZ6X62FB0mHel5lN
TEFjypy+PgdyIbRCU9z4FglCi2g1HuPptg0rNR/apu6MKKhFI5+X3nzA/l0UTkdI
E/tD7Fg/gbTuMowiFnM0iY0hC6AXom7GiUDwXfIr2cyGu8OkNUOSNy1WYWd2m3Nb
oSyjWA+RnBvYunDEwqsKzysJvC+umB88ApWm4enySPUAcwlP+vECAwEAAaOBnDCB
mTAMBgNVHRMEBTADAQH/MAsGA1UdDwQEAwIC9DA7BgNVHSUENDAyBggrBgEFBQcD
AQYIKwYBBQUHAwIGCCsGAQUFBwMDBggrBgEFBQcDBAYIKwYBBQUHAwgwEQYJYIZI
AYb4QgEBBAQDAgD3MCwGA1UdEQQlMCOGG2h0dHA6Ly9leGFtcGxlLm9yZy93ZWJp
ZCNtZYcEfwAAATANBgkqhkiG9w0BAQsFAAOCAQEAjDnIK8R7ruSiJpdUeZ2ZxaGI
ewf020sAc0ENw6x3jaKKFINNLnMqEKLntZzIg7WbjnKLhtUhb/SLHP59mfT2OJVS
YMW7uRmnajfEd0MzfrXIUKBwCeWTCyP/bcdlJOT4u24ngFnkUyWaEKnqH3YuI8cw
BKWQ26Lq7ODq6e6Otxf0KOtWTeVSzlCL66MGBvQ36LF1g8PLEZ9hvvX06ieaE99W
HDK9/pGSuobx6Fi7ufzqymirT/GOuxslquB4mleCu4ArCg2qAVC2wDjQymgvq8FS
bVukYnC6XdwqYhvIuTPnEys4gGdeirY+UPGQeqxMrNm+ZpKPTS0NwzxHwlhutw==
-----END CERTIFICATE-----`;
var uidai_offline_publickey_26022021 = `-----BEGIN CERTIFICATE-----
MIIHwjCCBqqgAwIBAgIEU5laMzANBgkqhkiG9w0BAQsFADCB/DELMAkGA1UEBhMC
SU4xQTA/BgNVBAoTOEd1amFyYXQgTmFybWFkYSBWYWxsZXkgRmVydGlsaXplcnMg
YW5kIENoZW1pY2FscyBMaW1pdGVkMR0wGwYDVQQLExRDZXJ0aWZ5aW5nIEF1dGhv
cml0eTEPMA0GA1UEERMGMzgwMDU0MRAwDgYDVQQIEwdHdWphcmF0MSYwJAYDVQQJ
Ex1Cb2Rha2RldiwgUyBHIFJvYWQsIEFobWVkYWJhZDEcMBoGA1UEMxMTMzAxLCBH
TkZDIEluZm90b3dlcjEiMCAGA1UEAxMZKG4pQ29kZSBTb2x1dGlvbnMgQ0EgMjAx
NDAeFw0yMTAyMjYxMTU0MjRaFw0yNDAyMjcwMDI3MTFaMIHdMQswCQYDVQQGEwJJ
TjExMC8GA1UEChMoVU5JUVVFIElERU5USUZJQ0FUSU9OIEFVVEhPUklUWSBPRiBJ
TkRJQTEPMA0GA1UEERMGMTEwMDAxMQ4wDAYDVQQIEwVEZWxoaTEbMBkGA1UECRMS
QkVISU5EIEtBTEkgTUFORElSMSQwIgYDVQQzExtBQURIQVIgSFEgQkFOR0xBIFNB
SElCIFJPQUQxNzA1BgNVBAMTLkRTIFVOSVFVRSBJREVOVElGSUNBVElPTiBBVVRI
T1JJVFkgT0YgSU5ESUEgMDUwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIB
AQCiciwOXy3lunB+2T8DbsKx8LlVkyOQ+swPC8vyDIChXAiLSIaGa3LrJasL9Vov
4Gtp7b1cyDt0x3CdshQebAfGi834WdPa9/P87SQdByBV3BVIhHS0XCyYL6lUqlKq
b/+ySBhhxlCF2EtkFY6fQ9nzXKabSM6TAFIhAqTK4JO//UdLCNMtHQQG9of35VvS
JqI4S/WKQcOEw5dPHHxRFYGckm3jrfPsu5kExIbx9dUwOXe+pjWENnMptcFor9yV
Ehcx9/SNQ6988x9pseO755Sdx6ixDAvd66ur3r6gdqHPgWat8GqKQd7fFDv/g129
K9W7C2HSRywjSm1EEbybU2CVAgMBAAGjggNnMIIDYzAOBgNVHQ8BAf8EBAMCBsAw
KgYDVR0lBCMwIQYIKwYBBQUHAwQGCisGAQQBgjcKAwwGCSqGSIb3LwEBBTCCAQIG
A1UdIASB+jCB9zCBhgYGYIJkZAICMHwwegYIKwYBBQUHAgIwbgxsQ2xhc3MgMiBj
ZXJ0aWZpY2F0ZXMgYXJlIHVzZWQgZm9yIGZvcm0gc2lnbmluZywgZm9ybSBhdXRo
ZW50aWNhdGlvbiBhbmQgc2lnbmluZyBvdGhlciBsb3cgcmlzayB0cmFuc2FjdGlv
bnMuMGwGBmCCZGQKATBiMGAGCCsGAQUFBwICMFQMUlRoaXMgY2VydGlmaWNhdGUg
cHJvdmlkZXMgaGlnaGVyIGxldmVsIG9mIGFzc3VyYW5jZSBmb3IgZG9jdW1lbnQg
c2lnbmluZyBmdW5jdGlvbi4wDAYDVR0TAQH/BAIwADAjBgNVHREEHDAagRhyYWh1
bC5rdW1hckB1aWRhaS5uZXQuaW4wggFuBgNVHR8EggFlMIIBYTCCAR6gggEaoIIB
FqSCARIwggEOMQswCQYDVQQGEwJJTjFBMD8GA1UEChM4R3VqYXJhdCBOYXJtYWRh
IFZhbGxleSBGZXJ0aWxpemVycyBhbmQgQ2hlbWljYWxzIExpbWl0ZWQxHTAbBgNV
BAsTFENlcnRpZnlpbmcgQXV0aG9yaXR5MQ8wDQYDVQQREwYzODAwNTQxEDAOBgNV
BAgTB0d1amFyYXQxJjAkBgNVBAkTHUJvZGFrZGV2LCBTIEcgUm9hZCwgQWhtZWRh
YmFkMRwwGgYDVQQzExMzMDEsIEdORkMgSW5mb3Rvd2VyMSIwIAYDVQQDExkobilD
b2RlIFNvbHV0aW9ucyBDQSAyMDE0MRAwDgYDVQQDEwdDUkw1Njk0MD2gO6A5hjdo
dHRwczovL3d3dy5uY29kZXNvbHV0aW9ucy5jb20vcmVwb3NpdG9yeS9uY29kZWNh
MTQuY3JsMCsGA1UdEAQkMCKADzIwMjEwMjI2MTE1NDI0WoEPMjAyNDAyMjcwMDI3
MTFaMBMGA1UdIwQMMAqACE0HvvGenfu9MB0GA1UdDgQWBBTpS5Cfqf2zdwqjupLA
qMwk/bqX9DAZBgkqhkiG9n0HQQAEDDAKGwRWOC4xAwIDKDANBgkqhkiG9w0BAQsF
AAOCAQEAbTlOC4sonzb44+u5+VZ3wGz3OFg0uJGsufbBu5efh7kO2DlYnx7okdEf
ayQQs6AUzDvsH1yBSBjsaZo3fwBgQUIMaNKdKSrRI0eOTDqilizldHqj113f4eUz
U2j4okcNSF7TxQWMjxwyM86QsQ6vxZK7arhBhVjwp443+pxfSIdFUu428K6yH4JB
GhZSzWuqD6GNhOhDzS+sS23MkwHFq0GX4erhVfN/W7XLeSjzF4zmjg+O77vTySCN
e2VRYDrfFS8EAOcO4q7szc7+6xdg8RlgzoZHoRG/GqUp9inpJUn7OIzhHi2e8Mll
aMdtXo0nbr150tMe8ZSvY2fMiTCY1w==
-----END CERTIFICATE-----
`;
var uidai_offline_publickey_17022026 = `-----BEGIN CERTIFICATE-----
MIIHCjCCBfKgAwIBAgIEYklh9TANBgkqhkiG9w0BAQsFADCBkTELMAkGA1UEBhMC
SU4xQTA/BgNVBAoTOEd1amFyYXQgTmFybWFkYSBWYWxsZXkgRmVydGlsaXplcnMg
YW5kIENoZW1pY2FscyBMaW1pdGVkMQ8wDQYDVQQLEwZTdWItQ0ExLjAsBgNVBAMT
JShuKUNvZGUgU29sdXRpb25zIFN1Yi1DQSBmb3IgRFNDIDIwMjIwHhcNMjQwMjIx
MTIyMzE5WhcNMjYwMjE2MjIyNzQwWjCB8zELMAkGA1UEBhMCSU4xMTAvBgNVBAoT
KFVOSVFVRSBJREVOVElGSUNBVElPTiBBVVRIT1JJVFkgT0YgSU5ESUExDzANBgNV
BBETBjExMDAwMTEOMAwGA1UECBMFRGVsaGkxODA2BgNVBAkTL0JhbmdsYSBTYWhp
YiBSb2FkIEJlaGluZCBLYWxpIE1hbmRpciBHb2xlTWFya2V0MR0wGwYDVQQzExRB
YWRoYWFyIEhlYWRxdWFydGVyczE3MDUGA1UEAxMuRFMgVW5pcXVlIElkZW50aWZp
Y2F0aW9uIEF1dGhvcml0eSBvZiBJbmRpYSAwNTCCASIwDQYJKoZIhvcNAQEBBQAD
ggEPADCCAQoCggEBAJjCCSo9vCXEzdQe9qdv6YMwwqILobP62yF4Df8mzTYdDVTq
2VdipZa5k+VTmJTKwAabyKeyVP5+oZmt/zteJ+5754Cf8t3KJYfK5O5rA8FQDu2b
lQ/hgs32mDyy5FEvsKh+zOxThyJ/6ciaNUZMZvfJTt3cqMeUbixtlKpuyX1SKJpt
Le0CiaDOa3/CTbz13cLMTfPXlpU9gX+wRf/xtxSohe2R37yMsqrXv96K0URvzPNE
Ki6wz9USHjmud2U4CC13MkDm9smTqjR94jdrkikbLrDTte06ORTm9ifR3ecCDLg7
988ODFk6ygFrk2KJ/dEEnhkSr9XeQEDUbdMeXUsCAwEAAaOCAwQwggMAMA4GA1Ud
DwEB/wQEAwIGwDAqBgNVHSUEIzAhBggrBgEFBQcDBAYKKwYBBAGCNwoDDAYJKoZI
hvcvAQEFMIIBAgYDVR0gBIH6MIH3MIGGBgZggmRkAgIwfDB6BggrBgEFBQcCAjBu
DGxDbGFzcyAyIGNlcnRpZmljYXRlcyBhcmUgdXNlZCBmb3IgZm9ybSBzaWduaW5n
LCBmb3JtIGF1dGhlbnRpY2F0aW9uIGFuZCBzaWduaW5nIG90aGVyIGxvdyByaXNr
IHRyYW5zYWN0aW9ucy4wbAYGYIJkZAoBMGIwYAYIKwYBBQUHAgIwVAxSVGhpcyBj
ZXJ0aWZpY2F0ZSBwcm92aWRlcyBoaWdoZXIgbGV2ZWwgb2YgYXNzdXJhbmNlIGZv
ciBkb2N1bWVudCBzaWduaW5nIGZ1bmN0aW9uLjAMBgNVHRMBAf8EAjAAMCIGA1Ud
EQQbMBmBF2Rpci5sY2ktaHFAdWlkYWkubmV0LmluMIIBAAYDVR0fBIH4MIH1MEKg
QKA+hjxodHRwOi8vd3d3Lm5jb2Rlc29sdXRpb25zLmNvbS9yZXBvc2l0b3J5L25j
b2RlY2EyMnN1YmNhMS5jcmwwga6ggauggaikgaUwgaIxCzAJBgNVBAYTAklOMUEw
PwYDVQQKEzhHdWphcmF0IE5hcm1hZGEgVmFsbGV5IEZlcnRpbGl6ZXJzIGFuZCBD
aGVtaWNhbHMgTGltaXRlZDEPMA0GA1UECxMGU3ViLUNBMS4wLAYDVQQDEyUobilD
b2RlIFNvbHV0aW9ucyBTdWItQ0EgZm9yIERTQyAyMDIyMQ8wDQYDVQQDEwZDUkwx
ODMwKwYDVR0QBCQwIoAPMjAyNDAyMjExMjIzMTlagQ8yMDI2MDIxNjIyMjc0MFow
HwYDVR0jBBgwFoAURsE2obZEOWzewDFm8UhoheJjvzswHQYDVR0OBBYEFKzhXAem
Tyikhe8ZRfMcBEnWfEfTMBkGCSqGSIb2fQdBAAQMMAobBFY4LjMDAgMoMA0GCSqG
SIb3DQEBCwUAA4IBAQAw7POBP18z1+IG4bjBk6vgfIyTH6Vz+e1j7jcmG+sUfY0E
GVKonznB87M4hNcVcylDrwZscJwZxKaYUAt/q1Uph46v37jRjZyAvawZBD/oL1lS
qUmIGQxI+zqUFYrc5dWSHROoCGfkyHAskr60Y3tN489NKo+AsFC1iu1dY29FW2hK
Wt1dQEDYkTeYJu8WPf/XaOR562gq566IFgL6cpYujNrMuKOGj+5o2UbjmCTjkbTp
HdTU+Mja3mYfUDdBia4YVQwDXB07xZD7FiMbhm1oYsATfjiFA1jCy3X8q9dsFkMe
tMjs/7td7cCA1L45savQhcL7+c902NZMolXgRnFx
-----END CERTIFICATE-----
`;

// src/verifySignature.ts
var verifyRSASha256WithSubtle = async (certificate, signature, signedData) => {
  const publicKey = import_node_forge.pki.certificateFromPem(certificate).publicKey;
  const publicKeyPem = import_node_forge.pki.publicKeyToPem(publicKey);
  const pemHeader = "-----BEGIN PUBLIC KEY-----";
  const pemFooter = "-----END PUBLIC KEY-----";
  const pemContents = publicKeyPem.substring(
    pemHeader.length,
    publicKeyPem.length - pemFooter.length - 2
  );
  const binaryDerString = window.atob(pemContents);
  const binaryDer = str2ab(binaryDerString);
  const pk = await window.crypto.subtle.importKey(
    "spki",
    binaryDer,
    { name: "RSASSA-PKCS1-v1_5", hash: "SHA-256" },
    true,
    ["verify"]
  );
  const isSignatureValid = await window.crypto.subtle.verify(
    { name: "RSASSA-PKCS1-v1_5" },
    pk,
    signature.buffer,
    signedData.buffer
  );
  return { isSignatureValid, certificate };
};
var verifySignature = async (qrData, useTestAadhaar) => {
  const bigIntData = BigInt(qrData);
  const byteArray = (0, import_core2.convertBigIntToByteArray)(bigIntData);
  const decompressedByteArray = (0, import_core2.decompressByteArray)(byteArray);
  const signature = decompressedByteArray.slice(
    decompressedByteArray.length - 256,
    decompressedByteArray.length
  );
  const signedData = decompressedByteArray.slice(
    0,
    decompressedByteArray.length - 256
  );
  if (useTestAadhaar) {
    const { isSignatureValid, certificate } = await verifyRSASha256WithSubtle(
      testCertificate,
      signature,
      signedData
    );
    return isSignatureValid ? { isSignatureValid, certificate } : { isSignatureValid };
  } else {
    const results = await Promise.all([
      verifyRSASha256WithSubtle(
        uidai_offline_publickey_26022021,
        signature,
        signedData
      ),
      verifyRSASha256WithSubtle(
        uidai_offline_publickey_17022026,
        signature,
        signedData
      )
    ]);
    const validResult = results.find((result) => result.isSignatureValid);
    return validResult ? { isSignatureValid: true, certificate: validResult.certificate } : { isSignatureValid: false };
  }
};

// src/prove.ts
var proveAndSerialize = async (anonAadhaarArgs, setProverState) => {
  let anonAadhaarProof;
  let serialized;
  try {
    anonAadhaarProof = await (0, import_core3.prove)(anonAadhaarArgs, setProverState);
    serialized = await (0, import_core3.serialize)(anonAadhaarProof);
  } catch (e) {
    console.error(e);
    throw new Error("Error while generating the proof");
  }
  return { anonAadhaarProof, serialized };
};
var processAadhaarArgs = async (qrData, useTestAadhaar, nullifierSeed, fieldsToRevealArray, signal) => {
  let certificateFile = null;
  try {
    if (useTestAadhaar) {
      return (0, import_core3.generateArgs)({
        qrData,
        certificateFile: testCertificate,
        nullifierSeed,
        fieldsToRevealArray,
        signal
      });
    } else {
      const { isSignatureValid, certificate } = await verifySignature(
        qrData,
        useTestAadhaar
      );
      if (!certificate)
        throw new Error(
          "[processAadhaarArgs]: Error while processing the arguments, no certificate retrieved"
        );
      if (isSignatureValid)
        certificateFile = certificate;
      if (!certificateFile)
        throw Error("Error while fetching public key.");
      return (0, import_core3.generateArgs)({
        qrData,
        certificateFile,
        nullifierSeed,
        fieldsToRevealArray,
        signal
      });
    }
  } catch (error) {
    if (error instanceof Error)
      throw new Error(error.message);
    throw new Error(JSON.stringify(error));
  }
};

// src/provider/AnonAadhaarProvider.tsx
var import_jsx_runtime = require("react/jsx-runtime");
function AnonAadhaarProvider(anonAadhaarProviderProps) {
  const [anonAadhaarProofStr, setAnonAadhaarProofStr] = (0, import_react2.useState)(null);
  const [anonAadhaarProof, setAnonAadhaarProof] = (0, import_react2.useState)(null);
  const [appName, setAppName] = (0, import_react2.useState)("The current application");
  const [proverState, setProverState] = (0, import_react2.useState)(
    import_core4.ProverState.Initializing
  );
  const [useTestAadhaar, setUseTestAadhaar] = (0, import_react2.useState)(false);
  const [state, setState] = (0, import_react2.useState)({
    status: "logged-out"
  });
  (0, import_react2.useEffect)(() => {
    if (anonAadhaarProviderProps._appName !== void 0)
      setAppName(anonAadhaarProviderProps._appName);
    if (anonAadhaarProviderProps._useTestAadhaar)
      setUseTestAadhaar(true);
  }, [
    anonAadhaarProviderProps._appName,
    anonAadhaarProviderProps._useTestAadhaar
  ]);
  (0, import_react2.useEffect)(() => {
    let anonAadhaarInitArgs;
    if (anonAadhaarProviderProps._artifactslinks) {
      anonAadhaarInitArgs = {
        wasmURL: anonAadhaarProviderProps._artifactslinks.wasm_url,
        zkeyURL: anonAadhaarProviderProps._artifactslinks.zkey_url,
        vkeyURL: anonAadhaarProviderProps._artifactslinks.vkey_url,
        artifactsOrigin: import_core4.ArtifactsOrigin.server
      };
    } else {
      anonAadhaarInitArgs = {
        wasmURL: import_core4.artifactUrls.v2.wasm,
        zkeyURL: import_core4.artifactUrls.v2.chunked,
        vkeyURL: import_core4.artifactUrls.v2.vk,
        artifactsOrigin: import_core4.ArtifactsOrigin.chunked
      };
    }
    (0, import_core4.init)(anonAadhaarInitArgs).then().catch((e) => {
      throw Error(e);
    });
  }, [anonAadhaarProviderProps._artifactslinks]);
  const setAndWriteState = (newState) => {
    console.log(`[ANON-AADHAAR] new state ${shallowToString(newState)}`);
    setState(newState);
    writeToLocalStorage(newState);
  };
  const startReq = import_react3.default.useCallback(
    (request) => {
      console.log(`[ANON-AADHAAR] startReq ${shallowToString(request)}`);
      setAndWriteState(
        handleLoginReq(
          request,
          state,
          setAnonAadhaarProofStr,
          setAnonAadhaarProof,
          setProverState
        )
      );
    },
    [setAndWriteState, setAnonAadhaarProofStr, setAnonAadhaarProof]
  );
  import_react3.default.useEffect(() => {
    if (anonAadhaarProofStr === null || anonAadhaarProof === null)
      return;
    console.log(`[ANON-AADHAAR] trying to log in with ${anonAadhaarProofStr}`);
    handleLogin(state, anonAadhaarProofStr, anonAadhaarProof, useTestAadhaar).then((newState) => {
      if (newState)
        setAndWriteState(newState);
      else
        console.log(
          `[ANON-AADHAAR] ${state.status}, ignoring anonAadhaarProof: ${anonAadhaarProofStr}`
        );
    }).catch((e) => {
      setAndWriteState({ status: "logged-out" });
      console.error(e);
      console.error(
        `[ANON-AADHAAR] error logging in, ignoring anonAadhaarProof: ${anonAadhaarProofStr}`
      );
    });
  }, [anonAadhaarProofStr]);
  const val = import_react3.default.useMemo(
    () => ({ state, startReq, proverState, appName, useTestAadhaar }),
    [state, proverState, appName, useTestAadhaar]
  );
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(AnonAadhaarContext.Provider, { value: val, children: anonAadhaarProviderProps.children });
}
function writeToLocalStorage(state) {
  console.log(`[ANON-AADHAAR] writing to local storage, status ${state.status}`);
  window.localStorage["anonAadhaar"] = serialize2(state);
}
function serialize2(state) {
  const { status } = state;
  let serState;
  if (status === "logged-out") {
    serState = {
      status: "logged-out"
    };
  } else {
    serState = {
      status,
      anonAadhaarProofs: state.anonAadhaarProofs
    };
  }
  return JSON.stringify(serState);
}
function shallowToString(obj) {
  return JSON.stringify(obj, function(key, val) {
    if (key === "")
      return val;
    if (val == null)
      return null;
    if (typeof val === "bigint")
      return "" + val;
    if (Array.isArray(val))
      return "<array>";
    if (typeof val === "object")
      return "<object>";
    return val;
  });
}
function handleLoginReq(request, state, setAnonAadhaarStr, setAnonAadhaar, setProverState) {
  const { type } = request;
  switch (type) {
    case "login":
      try {
        const { args } = request;
        proveAndSerialize(args, setProverState).then(
          ({
            anonAadhaarProof,
            serialized
          }) => {
            setAnonAadhaarStr(serialized);
            setAnonAadhaar(anonAadhaarProof);
          }
        );
      } catch (error) {
        console.log(error);
      }
      return {
        status: "logging-in",
        ...state.status !== "logged-out" ? {
          anonAadhaarProofs: state.anonAadhaarProofs
        } : {}
      };
    case "logout":
      return { status: "logged-out" };
    default:
      throw new Error(`Invalid request type ${type}`);
  }
}
async function handleLogin(state, _anonAadhaarProofStr, _anonAadhaarProof, useTestAadhaar) {
  if (state.status !== "logging-in") {
    console.log(
      `[ANON-AADHAAR] ignoring message. State != logging-in: ${state}`
    );
    return null;
  }
  if (!await (0, import_core4.verify)(_anonAadhaarProof, useTestAadhaar)) {
    throw new Error("Invalid proof");
  }
  const index = state.anonAadhaarProofs === void 0 ? 0 : Object.keys(state.anonAadhaarProofs).length;
  return {
    status: "logged-in",
    anonAadhaarProofs: {
      ...state.anonAadhaarProofs,
      [index]: _anonAadhaarProofStr
    }
  };
}

// src/components/AnonAadhaarProof.tsx
var import_react4 = require("react");
var import_styled_components = __toESM(require("styled-components"));
var import_jsx_runtime2 = require("react/jsx-runtime");
var AnonAadhaarProof = ({
  code,
  label
}) => {
  const [collapsed, setCollapsed] = (0, import_react4.useState)(true);
  const toggle = (0, import_react4.useCallback)(() => {
    setCollapsed((collapsed2) => !collapsed2);
  }, []);
  let buttonText = collapsed ? "Show proof" : "Hide proof";
  if (label !== void 0) {
    buttonText += " " + label;
  }
  if (collapsed) {
    return /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(RevealProofBtn, { onClick: toggle, children: buttonText });
  }
  return /* @__PURE__ */ (0, import_jsx_runtime2.jsxs)(import_jsx_runtime2.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(RevealProofBtn, { onClick: toggle, children: buttonText }),
    /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(ProofContainer, { children: /* @__PURE__ */ (0, import_jsx_runtime2.jsx)("pre", { children: code }) })
  ] });
};
var ProofContainer = import_styled_components.default.div`
  border-radius: 8px;
  border: 1px solid grey;
  overflow-y: scroll;
  max-width: 100%;
  padding: 8px;
`;
var RevealProofBtn = import_styled_components.default.button`
  font-size: 1rem;
  cursor: pointer;
  color: #000000;
  font-weight: bold;
  border-radius: 1.3125rem;
  background: #fff;
  box-shadow: 0px 3px 8px 1px rgba(0, 0, 0, 0.25);
  border: none;
  min-width: 8rem;
  min-height: 2rem;
  border-radius: 0.5rem;
`;

// src/components/LogInWithAnonAadhaar.tsx
var import_react12 = require("react");

// src/components/ProveModal/Modal.tsx
var import_react11 = require("react");
var import_styled_components9 = __toESM(require("styled-components"));

// src/components/ProveModal/ErrorToast.tsx
var import_react5 = require("react");
var import_styled_components2 = __toESM(require("styled-components"));
var import_jsx_runtime3 = require("react/jsx-runtime");
var ErrorToast = ({
  message,
  setErrorMessage
}) => {
  const [isVisible, setIsVisible] = (0, import_react5.useState)(false);
  (0, import_react5.useEffect)(() => {
    if (message)
      setIsVisible(true);
  }, [message]);
  const handleClose = () => {
    setIsVisible(false);
    setErrorMessage(null);
  };
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(import_jsx_runtime3.Fragment, { children: isVisible && /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)(Box, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(Message, { children: message }),
    /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(CloseButton, { onClick: handleClose })
  ] }) });
};
var Box = import_styled_components2.default.div`
  display: flex;
  position: absolute;
  width: 85%;
  border-radius: 5px;
  background-color: #efc8c8;
  padding: 4px;
  align-items: flex-start;
`;
var Message = import_styled_components2.default.div`
  text-align: left;
  margin-left: 3px;
  font-size: 0.875rem;
  color: #a00f0f;
  flex: 1;
  width: fit-content;
  max-width: 92%;
  overflow: hidden;
  text-overflow: ellipsis;
`;
var CloseButton = import_styled_components2.default.button`
  border: none;
  background: none;
  color: #a00f0f;
  cursor: pointer;
  width: 24px;
  height: 24px;
  display: flex;
  align-items: center;
  justify-content: center;

  &::before,
  &::after {
    content: '';
    position: absolute;
    width: 2px;
    height: 12px;
    background-color: currentColor;
  }

  &::before {
    transform: rotate(45deg);
  }

  &::after {
    transform: rotate(-45deg);
  }
`;

// src/components/ProveModal/Modal.tsx
var import_react_device_detect = require("react-device-detect");

// src/components/ProveModal/VerifyModal.tsx
var import_styled_components4 = __toESM(require("styled-components"));

// src/components/FileInput.tsx
var import_react6 = require("react");
var import_styled_components3 = __toESM(require("styled-components"));

// src/components/MainIcons.ts
var icons = {
  aalogo: '<svg xmlns="http://www.w3.org/2000/svg" width="173" height="173" fill="none"><path fill="#FD8B0E" d="M80.39.81v22.03c0 .45-.36.81-.81.81H31.52c-4.2 0-7.6 3.4-7.6 7.6V79.7c0 .45-.36.81-.81.81H.81c-.45 0-.81-.36-.81-.81V53.47c0-3.69 1.47-7.24 4.08-9.85l39.5-39.49c2.6-2.6 6.13-4.07 9.82-4.08L79.57 0c.45 0 .82.36.82.81"/><path fill="#081224" d="M91.85.81v22.03c0 .45.36.81.81.81h48.05c4.2 0 7.6 3.4 7.6 7.6V79.7c0 .45.36.81.81.81h22.29c.45 0 .81-.36.81-.81V53.47c0-3.69-1.47-7.24-4.08-9.85L128.66 4.13a13.94 13.94 0 0 0-9.82-4.08L92.67 0c-.45 0-.82.36-.82.81M80.39 171.56v-22.03c0-.45-.36-.81-.81-.81H31.52c-4.2 0-7.6-3.4-7.6-7.6V92.67c0-.45-.36-.81-.81-.81H.81c-.45 0-.81.36-.81.81v26.23c0 3.69 1.47 7.24 4.08 9.85l39.5 39.5c2.61 2.61 6.14 4.07 9.82 4.08l26.18.05c.45 0 .82-.36.82-.81z"/><path fill="#009A08" d="M91.85 171.56v-22.03c0-.45.36-.81.81-.81h48.05c4.2 0 7.6-3.4 7.6-7.6V92.67c0-.45.36-.81.81-.81h22.29c.45 0 .81.36.81.81v26.23c0 3.69-1.47 7.24-4.08 9.85l-39.5 39.5a13.934 13.934 0 0 1-9.82 4.08l-26.18.05c-.45 0-.82-.36-.82-.81z"/></svg>',
  illustration: '<svg xmlns="http://www.w3.org/2000/svg" width="82" height="82" fill="none"><path fill="#191919" d="M46.733 19.72c-1.386-2.03-4.097-1.749-6.242-1.734-5 .033-25.743.008-27.2.08-2.244.116-4.497 1.025-4.861 3.258-.197 1.206-.022 2.496.023 3.712.18 4.921.316 11.01.922 15.395.164 1.193.38 2.388 1.122 3.374 1.171 1.556 3.492 2.103 5.44 2.136.022.038 17.84.457 24.28-.025 1.875-.14 4.156-.282 5.42-1.885.633-.805.686-1.964.813-2.94.456-3.504.76-9.433 1.158-15.791.114-1.816.202-4.002-.875-5.58m.492 4.424c-.013.727-.555 9.325-.59 9.846-.12 1.764-.509 7.313-1.01 9.046-.496 1.71-2.724 2.131-4.238 2.292a51.503 51.503 0 0 1-3.863.26c-6.62.214-20.998.17-21.574.154-2.332-.108-4.949-.791-5.828-3.211a13.11 13.11 0 0 1-.534-3.384c-.438-5.278-.37-6.87-.707-14.994-.048-1.163-.285-2.53.248-3.618 1.17-2.376 4.702-2.043 6.866-2.067 2.545-.028 24.146-.044 25.483-.055 1.189-.01 2.472-.07 3.58.446 1.996.93 2.202 3.345 2.167 5.283zM28.015 47.531c.312.704 1.48.348 1.072-.592a.592.592 0 0 0-.879-.258c-.248.17-.465.627-.185.87a.08.08 0 0 0 .124-.012.08.08 0 0 0 .014-.045c.017-.186-.02-.384.135-.523a.368.368 0 0 1 .574.061.431.431 0 0 1-.075.583.49.49 0 0 1-.65-.16c-.051-.084-.168-.01-.13.076M26.426 51.863c.628.107 1.264.15 1.9.13.694.02 1.392.044 2.086.04a.178.178 0 0 0 0-.357c-.694-.028-1.392-.03-2.086-.033a9.359 9.359 0 0 0-1.9.063.083.083 0 0 0-.042.128c.01.014.026.024.042.03"/><path fill="#191919" d="M49.812 18.044a4.262 4.262 0 0 0-2.337-2.22c-1.325-.52-2.8-.438-4.195-.425-5.942.055-30.336.037-32.016.106a7.593 7.593 0 0 0-3.727 1.034c-2.533 1.6-1.955 4.27-1.854 7.262.195 5.755.343 12.898.988 18.013.17 1.347.357 2.71 1.08 3.891 1.292 2.106 3.908 2.901 6.369 2.891.095 0 5.565.131 11.839.178.01.331.075.697.112 1.046-.137-.017-3.82.075-4.903.23a.156.156 0 0 0-.113.21c-.555.47-1.046 1.051-1.552 1.564-.622.628-1.28 1.237-1.86 1.903a.133.133 0 0 0 .187.188c.652-.598 1.244-1.27 1.857-1.907.51-.532 1.094-1.046 1.554-1.627a45.948 45.948 0 0 1 4.856-.175c.003.264-.011.577.251.71.04.022.11.007.118-.048.02-.35.005-.7-.043-1.046a3.288 3.288 0 0 0-.195-1.045c1.462.01 2.964.015 4.449.013a3.567 3.567 0 0 0-.08 1.859.095.095 0 0 0 .18 0 2.31 2.31 0 0 0 .093-.538c.823.11 1.504.033 3.721.172.337.02.817.026 1.1.024a8.072 8.072 0 0 0 1.461 1.538c.541.532 1.111 1.033 1.707 1.501.144.109.322-.129.193-.25-.574-.524-1.177-1.015-1.75-1.543a15.075 15.075 0 0 0-1.395-1.314.111.111 0 0 0-.048-.13 2.664 2.664 0 0 0-1.063-.167c-.48-.03-.96-.054-1.44-.07-.819-.081-1.643-.11-2.466-.086.023-.357.084-.686.087-.996 4.236-.01 8.298-.083 10.846-.285 2.163-.172 4.81-.298 6.29-2.132.767-.948.839-2.26.99-3.425.635-4.92.901-12.521 1.303-18.152.162-2.21.352-4.681-.594-6.752m.26 5.067c-.023.68-.62 10.508-.683 11.488-.136 2.065-.582 8.506-1.144 10.524-.543 1.95-2.943 2.52-4.684 2.732a51.73 51.73 0 0 1-4.504.328c-7.638.26-24.618.165-24.922.154-2.77-.121-5.716-1.018-6.666-3.912-.42-1.28-.471-2.677-.578-4.013-.42-5.297-.391-6.99-.764-17.321-.05-1.37-.293-2.89.097-4.226.917-3.134 5.046-2.929 7.593-2.962 3.013-.04 28.773-.071 30.311-.08 1.386-.008 2.879.027 4.053.865 1.944 1.386 1.967 4.277 1.892 6.422zM3.215 64.738c.243-.345 5.932-6.756 6.215-7.075a1.717 1.717 0 0 1 1.467-.603c3.875-.023 7.751-.047 11.626.015 7.231.118 23.137-.287 23.67-.282.523.006 1.198-.115 1.597.269.38.366 4.316 5.719 5.691 7.152.214.223.552-.115.339-.338-1.347-1.405-4.71-5.932-5.259-6.616a2.778 2.778 0 0 0-.764-.812c-.686-.342-8.65.075-12.054.153-4.03.092-2.993.02-24.419.015-.948 0-1.619.04-2.243.816-.718.893-5.823 6.814-6.052 7.196-.071.122.107.22.186.11M53.128 66.26c.351-.598 1.079-1.576.73-1.689-.54-.172-50.546.289-50.546.289s-.962.025-.04.883c.922.857.965 1.077 2.63 1.012 1.664-.064 42.49-.194 43.629-.176 1.138.018 3.374.06 3.597-.319"/><path fill="#191919" d="M8.526 60.692c-.354.438-1.925 2.053-1.989 2.453a.174.174 0 0 0 .077.188c.214.138 6.734 0 8.187-.005 3.07-.013 24.83.12 25.381.128.421.005.72-.084.513-.556-.703-1.608-1.402-3.217-2.116-4.82-.182-.407-.327-.462-.764-.46 0 0-26.107 0-26.933-.01a.149.149 0 0 0 0 .298c5.161.05 25.73-.014 27.178.072.117.007-.04-.265 2.258 4.918.033.072.024.201-.357.21-.264.006-25.514-.1-31.297-.057-.308 0-1.611.054-1.602-.023.032-.259 2.042-2.395 3.757-5.022.059-.09-.08-.165-.142-.083-.003-.001-1.366 1.796-2.151 2.769M41.396 62.223c.132.387.301.76.505 1.115.12.166.201.164.402.17 1.122.035 4.24-.107 7.36-.046.206.004 1.057.134 1.045-.201-.016-.492-1.511-2.231-3.573-5.448-.235-.368-.336-.39-.802-.37-.76.032-5.242.194-6.721.242-.197.006-.198.312 0 .306 7.633-.241 7.045-.26 7.092-.23.15.096.301.466.398.614.215.327 2.834 4.225 3.071 4.566.142.203.003.191-.281.186-.322-.007-7.309.038-7.536.036-.281-.004-.419-.553-.534-.798-1.474-3.151-1.71-3.637-2.112-4.294-.064-.104-.228-.072-.183.033.705 1.651 1.228 2.67 1.869 4.119"/><path fill="#F2AE7F" d="M23.74 26.961c-.133-1.717.603-3.482 1.326-4.222a4.21 4.21 0 0 1 5.554-.171 5.759 5.759 0 0 1 1.92 3.575c.078.329.361 1.16-.764 1.236-.08.006-.311-.197-.343-.26-.322-.623-.376-1.343-.648-1.988a3.582 3.582 0 0 0-1.725-1.896 2.417 2.417 0 0 0-2.48.232 3.69 3.69 0 0 0-1.189 2.232c-.367 2.346.55 4.007.783 4.107a.983.983 0 0 0 .412.05c1.106-.027 5.503-.201 6.55-.18.111.003.358.04.428.233.25.694.108 2.377.095 2.55-.054.663-.46.233-1.14.66-2.272-.281-3.067-.394-3.499-.27a.676.676 0 0 0-.426.32.694.694 0 0 0 .408.876c.33.146.727.192.992.335-.022.442.03.61.009 1.055a.77.77 0 0 0 .032.332.514.514 0 0 0 .41.281c.173.015.348 0 .516-.044a4.75 4.75 0 0 1 .273-1.625c.334.055.708.191 1.042.247.04.009-.293 1.113-.23 1.559.022.158.287.115.313.147.39.495.54.597.81.662.303.074.349.212.406.575.402 2.563-2.19 4.877-4.859 4.923-1.495.027-4.084 0-5.957-2.702a3.587 3.587 0 0 1-.684-2.017c.048-.174.128-.337.235-.482.647-.618 1.153-1.153 1.066-2.283-.072-.937-.456-1.786-1.35-1.953-.078-.014-.254-.28-.266-.358a9.533 9.533 0 0 1-.057-2.506.357.357 0 0 1 .156-.15c.504-.107 1.778-.033 2.279-.127a.49.49 0 0 0 .134-.38 9.466 9.466 0 0 1-.533-2.573"/><path fill="#191919" d="M38.797 29.258c-.046.755-.507 9.873-.32 10.614a.1.1 0 0 0 .098.072.102.102 0 0 0 .097-.072c.216-.703.222-9.724.21-10.614a.043.043 0 0 0-.042-.043.043.043 0 0 0-.043.043M58.86 32.705a4.651 4.651 0 0 0-3.388.274c-1.367.668-2.26 2.064-2.794 3.547a1.563 1.563 0 0 0-.792-.668c-.599-.223-1.794 0-1.827.796a.12.12 0 0 0 .206.084c.42-.336.876-.68 1.448-.467a1.06 1.06 0 0 1 .68.82 1.136 1.136 0 0 1-.647 1.127 1.389 1.389 0 0 1-1.609-.556c-.02-.024-.068-.02-.064.016.084.903 1.352 1.287 2.096.793-.379 2.206.095 4.806 2.095 5.823a4.169 4.169 0 0 0 5.692-1.916c.127-.262-.26-.492-.394-.23a3.703 3.703 0 0 1-4.581 1.922c-1.999-.644-2.622-2.922-2.466-4.81.119-1.211.457-2.39.999-3.48.37-.88.986-1.636 1.775-2.174a4.352 4.352 0 0 1 3.173-.526c1.751.377 2.778 2.431 2.618 4.078-.009.086.116.09.142.018.624-1.71-.696-3.897-2.363-4.47"/><path fill="#191919" d="M58.41 41.094c-.608.469-1.2.997-2.005 1.045-1.422.087-2.805-1.633-1.835-4.208.478-1.269 1.522-3.032 3.051-3.08a1.76 1.76 0 0 1 1.61.907c.197.39.092.801.234 1.19.04.11.223.17.283.037.385-.847-.208-1.89-.956-2.308-1.126-.629-2.408-.12-3.214.775-1.477 1.644-2.843 5.164-.486 6.704 1.177.77 2.633.087 3.428-.917.061-.073-.02-.216-.11-.145M29.34 32.979a4.936 4.936 0 0 1 1.718.086c2.65.327 5.288 1.008 6.985.893.062-.005.087-.103.015-.117-1.176-.224-2.386-.26-3.568-.464-1.132-.195-2.26-.418-3.398-.57a6.135 6.135 0 0 0-1.772-.142 1.126 1.126 0 0 0-1.046-.47c-1.352.01-1.642 1.475-1.134 2.342.16.273.364.52.603.729a.314.314 0 0 1 .09.348 39.27 39.27 0 0 1-.774 1.94.175.175 0 0 0 .012.163.173.173 0 0 0 .142.08c.79.022 1.613.053 2.401.036a.171.171 0 0 0 .167-.154.173.173 0 0 0-.01-.08c-.27-.695-.42-1.344-.696-2.014a.307.307 0 0 1 .088-.358 1.104 1.104 0 0 0 .378-.883c.644.246-.07 1.184.175 1.636.141.26.831.368 1.092.305.466-.114.338-1.304.49-1.557.093-.158.753.047.777.09a1.64 1.64 0 0 1-.076.373c-.066.361-.362 1.012.092 1.175.173.028.348.043.523.044.134.042.273.07.413.086.45-.035.344-.729.418-1.081.104-.503.364-.37.773-.294a37.46 37.46 0 0 0 3.71.634c.169.014.211-.275.04-.301-1.006-.152-1.998-.3-2.991-.519a6.425 6.425 0 0 0-1.453-.295c-.3.013-.313.23-.362.491a3.217 3.217 0 0 1-.143.882c-.125.186-.306.08-.506.062-.48-.043-.305-.279-.294-.344.044-.263.306-.942.066-1.163a2.648 2.648 0 0 0-1.006-.227c-.17.026-.197.11-.246.257-.057.22-.102.441-.135.665-.029.137.033.799-.5.644-.3-.118-.338-.161-.373-.487-.036-.326.148-.676.11-.996-.03-.246-.103-.268-.349-.326-.264-.062-.574-.035-.795-.22-.444-.372-.053-.796.349-.899M49.026 33.319v.071c0 .006.01.006.01 0 0-.023 0-.048.004-.071 0-.008-.014-.009-.014 0M17.623 54.048s21.258-.274 21.464-.23c.206.046-.05 1.42-.684 1.417-.633-.004-19.623.149-19.93.027-.307-.121-1.178-1.21-.85-1.214M61.926 35.055c.718.008 5.775-.568 6.485-.495.727.073 10.023 1.952 10.746 1.92.055 0 .09-.087.031-.114-.672-.306-9.827-2.09-10.686-2.168-.74-.064-5.3.285-6.584.791a.033.033 0 0 0 .008.066"/><path fill="#191919" d="M58.777 38.473c5.509.25 8.002-2.433 11.887-1.649.172.035.414.033.45-.138.04-.182-.209-.255-.39-.295a7.166 7.166 0 0 0-4.377.272 26.14 26.14 0 0 1-7.57 1.73.04.04 0 0 0-.03.068.04.04 0 0 0 .03.012"/><path fill="#191919" d="M59.245 39.578c1.792 2.044 5.152 1.198 7.425.8 2.237-.393 4.046.607 6.213.885.757.098 1.717.181 2.423-.17.126-.064.047-.225-.065-.239-.483-.059-1.22.189-2.797-.137-.975-.202-1.884-.563-2.838-.822-2.395-.643-4.745.764-7.114.569-2.89-.236-3.795-2.01-3.852-2.01.135.406.34.787.605 1.124M60.363 43.519c1.665.085 3.335.167 4.987.395 2.301.319 4.488 1.37 6.843 1.229 2.356-.14 4.592-1.096 6.963-1.116a.126.126 0 0 0 .064-.233c-1.254-.652-2.319.007-4.026.348-1.459.29-2.875.692-4.359.494-1.83-.245-3.559-.954-5.397-1.185a19.28 19.28 0 0 0-5.092-.043c-.063.011-.043.108.017.11M10.286 62.706l.003-.001v-.006l-.003-.001-.002.001v.006zM38.604 61.193c-.361-.006-.97-.01-1.775-.013-.194-.485.084.236-.7-1.741.705 0 1.245 0 1.575.004a.125.125 0 0 0 0-.25c-.338-.004-.916-.009-1.679-.012-.183-.442-.243-.51-.39-.588a.09.09 0 0 0-.11.017.09.09 0 0 0-.019.034c-.047.135-.053.217.058.535-1.41-.005-3.318-.007-5.45-.008a3.787 3.787 0 0 0-.326-.72.083.083 0 0 0-.154.04c.005.229.032.456.08.679-1.814 0-3.774 0-5.717.004l-.03-.513a.095.095 0 0 0-.094-.088.094.094 0 0 0-.094.088c-.017.171-.035.343-.05.514-2.026.002-4.022.007-5.81.012a1.032 1.032 0 0 0-.06-.459c-.018-.043-.075-.071-.11-.03a1.02 1.02 0 0 0-.214.49c-2.034.007-3.766.016-4.908.026.117-.182.215-.376.29-.58.061-.17-.197-.328-.315-.182-.199.249-.387.505-.566.768-.273.001-.547.006-.82.015a.082.082 0 0 0-.058.14.082.082 0 0 0 .058.023c.187.014.413.027.673.038-.955 1.42-.877 1.267-1.218 1.782a48.47 48.47 0 0 0-1.105.016.081.081 0 0 0-.082.082.082.082 0 0 0 .082.082c.225.016.55.03.961.04a7.85 7.85 0 0 0-.54.986.07.07 0 0 0 .078.102.07.07 0 0 0 .042-.032c.241-.35.51-.702.786-1.046 1.368.031 3.466.044 5.915.046a3.712 3.712 0 0 0-.21.953c-.007.117.145.133.202.054.21-.308.372-.647.479-1.005 1.955 0 4.11-.005 6.284-.013-.04.341-.002.686.113 1.01a.14.14 0 0 0 .19.05.14.14 0 0 0 .051-.05c.129-.321.181-.667.154-1.012 2.14-.008 4.284-.019 6.256-.028.087.314.211.617.37.902a.126.126 0 0 0 .233-.063 3.954 3.954 0 0 0-.132-.84c2.198-.01 4.159-.019 5.63-.019.134.371.317.723.544 1.046.057.076.22.066.208-.056a4.194 4.194 0 0 0-.252-.99c.719 0 1.29.003 1.678.01a.126.126 0 0 0 .115-.078.124.124 0 0 0-.117-.172m-27.523.022c.509-.624 1.02-1.223 1.386-1.756 1.226.04 2.97.058 4.979.065-.146.564-.323 1.12-.53 1.664-2.431.008-4.505.017-5.835.027m6.293-.028c.38-1.275.419-1.287.497-1.662 1.807.005 3.807 0 5.825-.01-.09 1.012-.061 1.057-.112 1.659-2.07.003-4.14.007-6.21.013m6.683-.014c-.023-.537.005-.683-.042-1.659 1.97-.009 3.946-.023 5.771-.036.1.393-.017-.103.46 1.69-1.952.001-4.073.002-6.189.005m6.66-.003-.504-1.695c2.141-.015 4.048-.029 5.447-.035.314.824.215.508.67 1.738a1774.75 1774.75 0 0 0-5.612-.008M48.372 61.506c-.216-.077-.409-.108-1.347-.096a22.81 22.81 0 0 0-1.212-2.085c.401.022.803-.014 1.194-.106a.127.127 0 0 0 0-.242 4.396 4.396 0 0 0-1.47-.073 33.517 33.517 0 0 0-.402-.588.109.109 0 0 0-.188.11l.274.49c-.67.024-1.755.07-2.466.102a3.967 3.967 0 0 0-.595-.74.072.072 0 0 0-.085-.01.072.072 0 0 0-.033.078c.066.239.153.47.26.693a7.32 7.32 0 0 0-1.429.123.082.082 0 0 0 0 .159c.532.11 1.077.139 1.618.087.215.395.458.78.66 1.153.163.3.316.622.48.938a6.609 6.609 0 0 0-1.542.116.08.08 0 0 0-.065.08.08.08 0 0 0 .065.078c.574.108 1.16.14 1.743.098.16.294.352.57.573.822.06.066.155-.006.14-.08a3.796 3.796 0 0 0-.235-.742c.855-.004 1.986-.014 2.534-.022.126.227.27.444.43.649.062.08.208.04.175-.073a4.88 4.88 0 0 0-.217-.58c.384.027.77-.006 1.144-.097a.126.126 0 0 0-.004-.242m-4.243-.02c-.196-.39-.425-.769-.625-1.131-.167-.301-.335-.633-.52-.956.603-.014 1.401-.036 2.47-.065.582 1.045.922 1.672 1.147 2.084-.532.008-1.053.026-2.472.066z"/></svg>',
  alert: '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path fill="#fff" fill-rule="evenodd" d="M12 4a8 8 0 1 0 0 16 8 8 0 0 0 0-16M2 12C2 6.477 6.477 2 12 2s10 4.477 10 10-4.477 10-10 10S2 17.523 2 12" clip-rule="evenodd"/><path fill="#fff" d="M13 8a1 1 0 1 0-2 0v4a1 1 0 1 0 2 0zM12 15a1 1 0 1 0 0 2 1 1 0 0 0 0-2"/></svg>',
  arrowDown: '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path fill="#fff" fill-rule="evenodd" d="M5.293 9.707a1 1 0 0 1 1.32-1.497l.094.083L12 13.585l5.293-5.292a1 1 0 0 1 1.32-.083l.094.083a1 1 0 0 1 .083 1.32l-.083.094-6 6a1 1 0 0 1-1.32.083l-.094-.083z" clip-rule="evenodd"/></svg>',
  arrowUp: '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path fill="#fff" fill-rule="evenodd" d="M18.707 14.293a1 1 0 0 1-1.32 1.497l-.094-.083L12 10.415l-5.293 5.292a1 1 0 0 1-1.32.083l-.094-.083a1 1 0 0 1-.083-1.32l.083-.094 6-6a1 1 0 0 1 1.32-.083l.094.083z" clip-rule="evenodd"/></svg>',
  arrow: '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path fill="#fff" fill-rule="evenodd" d="m16.58 11.012-2.29-2.285v-.01a.987.987 0 0 1 0-1.418c.39-.399 1.02-.399 1.42 0l4 3.992c.236.23.33.554.28.86a1 1 0 0 1-.39.656L15.7 16.7c-.19.18-.45.289-.71.289L15 17c-.27 0-.53-.11-.71-.29a.996.996 0 0 1-.01-1.417v-.01l2.28-2.275H5c-.56 0-1-.45-1-.998a.99.99 0 0 1 1-.998z" clip-rule="evenodd"/></svg>',
  externalLink: '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path fill="#fff" fill-rule="evenodd" d="M20.383 3.076A.997.997 0 0 0 20 3h-5l-.117.007A1 1 0 0 0 14 4l.007.117A1 1 0 0 0 15 5h2.586l-8.293 8.293-.083.094a1 1 0 0 0 1.497 1.32L19 6.414V9l.007.117A1 1 0 0 0 21 9V4a1.001 1.001 0 0 0-.617-.924M11 6a1 1 0 0 1 .117 1.993L11 8H6a1 1 0 0 0-.993.883L5 9v9a1 1 0 0 0 .883.993L6 19h9a1 1 0 0 0 .993-.883L16 18v-5a1 1 0 0 1 1.993-.117L18 13v5a3 3 0 0 1-2.824 2.995L15 21H6a3 3 0 0 1-2.995-2.824L3 18V9a3 3 0 0 1 2.824-2.995L6 6z" clip-rule="evenodd"/></svg>',
  menu: '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path fill="#fff" fill-rule="evenodd" d="M20 5a1 1 0 0 1 .117 1.993L20 7H4a1 1 0 0 1-.117-1.993L4 5zm0 6a1 1 0 0 1 .117 1.993L20 13H4a1 1 0 0 1-.117-1.993L4 11zm1 7a1 1 0 0 0-1-1H4l-.117.007A1 1 0 0 0 4 19h16l.117-.007A1 1 0 0 0 21 18" clip-rule="evenodd"/></svg>',
  x: '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path fill="#fff" fill-rule="evenodd" d="M7.382 6.18a.857.857 0 0 0-1.13 1.283L10.787 12 6.25 16.537l-.071.08a.857.857 0 0 0 1.283 1.132L12 13.212l4.537 4.537.08.071a.857.857 0 0 0 1.132-1.283L13.212 12l4.537-4.537.071-.08a.857.857 0 0 0-1.283-1.132L12 10.788 7.463 6.25z" clip-rule="evenodd"/></svg>',
  fileUpload: '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path fill="#081224" fill-rule="evenodd" d="M17 22zc1.65 0 3-1.35 3-3V8a1.011 1.011 0 0 0-.29-.71l-5-5A.966.966 0 0 0 14 2H7C5.34 2 4 3.34 4 5v14c0 1.65 1.34 3 3 3zM13 4H7c-.56 0-1 .44-1 1v14c0 .55.44 1 1 1h10c.55 0 1-.45 1-1V9h-3a2 2 0 0 1-2-2zm2 1.41V7h1.59zM12 18c-.56 0-1-.45-1-1v-3.585L9.71 14.71v-.01c-.4.39-1.03.39-1.42 0-.4-.4-.4-1.03 0-1.42l3-3c.38-.4 1.02-.4 1.41-.01l3 3c.39.38.39 1.02 0 1.41h-.01c-.19.18-.45.29-.71.29L15 15c-.27 0-.53-.11-.71-.29L13 13.414V17c0 .55-.45 1-1 1" clip-rule="evenodd"/></svg>',
  xBlack: '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path fill="#334155" fill-rule="evenodd" d="M7.382 6.18a.857.857 0 0 0-1.13 1.283L10.787 12 6.25 16.537l-.071.08a.857.857 0 0 0 1.283 1.132L12 13.212l4.537 4.537.08.071a.857.857 0 0 0 1.132-1.283L13.212 12l4.537-4.537.071-.08a.857.857 0 0 0-1.283-1.132L12 10.788 7.463 6.25z" clip-rule="evenodd"/></svg>',
  eyeOff: '<svg xmlns="http://www.w3.org/2000/svg" width="18" height="18" fill="none"><path fill="#B6B9C3" d="M4.478 4.478c-1.54 1.159-2.39 2.71-2.776 3.577a2.315 2.315 0 0 0 0 1.89C2.356 11.41 4.42 15 9 15c1.906 0 3.395-.63 4.522-1.478l-1.074-1.074c-.872.61-1.997 1.052-3.448 1.052-3.641 0-5.332-2.83-5.927-4.165a.815.815 0 0 1 0-.67C3.418 7.89 4.166 6.55 5.48 5.603a.758.758 0 0 0 .067-.054z"/><path fill="#B6B9C3" d="M6.879 6.879a3 3 0 1 0 4.243 4.243L10.06 10.06A1.5 1.5 0 1 1 7.94 7.94zM9 3c4.582 0 6.646 3.59 7.298 5.055a2.316 2.316 0 0 1 0 1.89 9.937 9.937 0 0 1-.476.93.75.75 0 1 1-1.298-.75c.172-.298.305-.57.403-.79a.816.816 0 0 0 0-.67C14.333 7.33 12.642 4.5 9 4.5h-.75a.75.75 0 1 1 0-1.5z"/><path fill="#B6B9C3" fill-rule="evenodd" d="M1.72 1.72a.75.75 0 0 1 1.06 0l13.5 13.5a.75.75 0 1 1-1.06 1.06L1.72 2.78a.75.75 0 0 1 0-1.06" clip-rule="evenodd"/></svg>',
  eye: '<svg xmlns="http://www.w3.org/2000/svg" width="18" height="18" fill="none"><path fill="#009A08" fill-rule="evenodd" d="M9 4.5c-3.641 0-5.332 2.83-5.927 4.165a.816.816 0 0 0 0 .67C3.668 10.67 5.359 13.5 9 13.5c3.642 0 5.333-2.83 5.927-4.165a.816.816 0 0 0 0-.67C14.333 7.33 12.642 4.5 9 4.5M1.703 8.055C2.355 6.59 4.419 3 9 3c4.582 0 6.646 3.59 7.298 5.055a2.316 2.316 0 0 1 0 1.89C15.646 11.409 13.582 15 9 15c-4.581 0-6.645-3.59-7.297-5.055a2.316 2.316 0 0 1 0-1.89" clip-rule="evenodd"/><path fill="#009A08" fill-rule="evenodd" d="M9 6a3 3 0 1 0 0 6 3 3 0 0 0 0-6m-.064 1.501L9 7.5a1.5 1.5 0 1 1-1.499 1.436 1.125 1.125 0 0 0 1.435-1.435" clip-rule="evenodd"/></svg>'
};

// src/components/FileInput.tsx
var import_jsx_runtime4 = require("react/jsx-runtime");
var FileInput = ({
  onChange,
  id,
  setQrStatus,
  qrStatus
}) => {
  const [fileName, setFileName] = (0, import_react6.useState)(null);
  const fileInputRef = (0, import_react6.useRef)(null);
  const uploadIcon = createBlobURL(icons.fileUpload);
  const xIcon = createBlobURL(icons.xBlack);
  const clearFileInput = () => {
    if (fileInputRef.current) {
      fileInputRef.current.value = "";
      setFileName(null);
      setQrStatus(null);
    }
  };
  return /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(import_jsx_runtime4.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(InputFile, { htmlFor: id, children: [
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
        "input",
        {
          type: "file",
          id,
          ref: fileInputRef,
          onChange: (e) => {
            if (!e.target.files)
              return;
            setFileName(e.target.files[0].name);
            onChange(e);
          },
          accept: "image/*",
          hidden: true
        }
      ),
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(FileUploadIcon, { src: uploadIcon }),
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(FileName, { id: "file-chosen", children: "Choose file" })
    ] }),
    fileName && (() => {
      switch (qrStatus) {
        case "QR code invalid \u274C" /* ERROR_PARSING_QR */:
          return /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(import_jsx_runtime4.Fragment, { children: [
            /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(InputFileWrong, { children: [
              /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(FileName, { id: "file-chosen", children: fileName }),
              /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("button", { onClick: clearFileInput, children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(FileUploadIcon, { src: xIcon }) })
            ] }),
            /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(DocumentResultWrong, { children: "Invalid QR Code." })
          ] });
        case "Signature verified \u2705" /* SIGNATURE_VERIFIED */:
          return /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(import_jsx_runtime4.Fragment, { children: [
            /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(InputFileCorrect, { children: [
              /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(FileName, { id: "file-chosen", children: fileName }),
              /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("button", { onClick: clearFileInput, children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(FileUploadIcon, { src: xIcon }) })
            ] }),
            /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(DocumentResultCorrect, { children: "Valid QR Code." })
          ] });
        case "QR code scanned, verifying QR Code \u{1F50E}" /* QR_CODE_SCANNED */:
          return /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(import_jsx_runtime4.Fragment, { children: [
            /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(FileNameContainer, { children: [
              /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(FileName, { id: "file-chosen", children: fileName }),
              /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("button", { onClick: clearFileInput, children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(FileUploadIcon, { src: xIcon }) })
            ] }),
            /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(DocumentResult, { children: "Verifying QR Code." })
          ] });
      }
    })()
  ] });
};
var FileName = import_styled_components3.default.span`
  margin-left: 5px;
`;
var InputFile = import_styled_components3.default.label`
  display: flex;
  align-items: center;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 80%;
  border-radius: 4px;
  border-width: 1px;
  border-color: black;
  max-width: 100%;
  font-size: '16px';
  line-height: 1.25rem;
  color: #111827;
  padding-top: 6px;
  padding-bottom: 6px;
  padding-left: 14px;
  padding-right: 14px;
  cursor: pointer;
  margin-top: 0.3rem;
`;
var FileNameContainer = import_styled_components3.default.label`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  display: flex;
  align-items: center;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 80%;
  border-radius: 4px;
  border-width: 1px;
  border-color: black;
  max-width: 100%;
  font-size: '16px';
  line-height: 1.25rem;
  color: #111827;
  padding-top: 6px;
  padding-bottom: 6px;
  padding-left: 14px;
  padding-right: 14px;
  cursor: pointer;
  margin-top: 0.3rem;
`;
var InputFileCorrect = import_styled_components3.default.label`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  display: flex;
  align-items: center;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 80%;
  border-radius: 4px;
  border-width: 2px;
  border-color: #00bf06;
  max-width: 100%;
  font-size: '16px';
  line-height: 1.25rem;
  color: #111827;
  padding-top: 6px;
  padding-bottom: 6px;
  padding-left: 14px;
  padding-right: 14px;
  cursor: pointer;
  margin-top: 0.3rem;
`;
var InputFileWrong = import_styled_components3.default.label`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  display: flex;
  align-items: center;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 80%;
  border-radius: 4px;
  border-width: 2px;
  border-color: #ef4444;
  max-width: 100%;
  font-size: '16px';
  line-height: 1.25rem;
  color: #111827;
  padding-top: 6px;
  padding-bottom: 6px;
  padding-left: 14px;
  padding-right: 14px;
  cursor: pointer;
  margin-top: 0.3rem;
`;
var FileUploadIcon = import_styled_components3.default.img`
  height: 1.5rem;
`;
var DocumentResultCorrect = import_styled_components3.default.div`
  color: #00bf06;
  position: absolute;
  font-size: 0.875rem;
  margin-top: 4px;
`;
var DocumentResult = import_styled_components3.default.div`
  color: #717686;
  position: absolute;
  font-size: 0.875rem;
  margin-top: 4px;
`;
var DocumentResultWrong = import_styled_components3.default.div`
  color: #ef4444;
  position: absolute;
  font-size: 0.875rem;
  margin-top: 4px;
`;

// src/hooks/useFonts.ts
var import_react7 = require("react");
var useFonts = () => {
  (0, import_react7.useEffect)(() => {
    if (window) {
      const element = document.createElement("link");
      element.setAttribute("rel", "stylesheet");
      element.setAttribute("type", "text/css");
      element.setAttribute(
        "href",
        "https://fonts.googleapis.com/css2?family=Rajdhani:wght@300;400;500;600;700&display=swap"
      );
      document.getElementsByTagName("head")[0].appendChild(element);
    }
  }, []);
};

// src/components/ProveModal/VerifyModal.tsx
var import_jsx_runtime5 = require("react/jsx-runtime");
var VerifyModal = ({
  qrStatus,
  provingEnabled,
  setQrStatus,
  setQrData,
  setCurrentView,
  useTestAadhaar
}) => {
  useFonts();
  return /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)(MainContainer, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)(Container, { children: [
      /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(TitleSection, { children: /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)(Disclaimer, { children: [
        "Anon Aadhaar allows you to create a proof of your Aadhaar ID without revealing any personal data. This process is local to your browser for privacy, and QR images are not uploaded to any server.",
        /* @__PURE__ */ (0, import_jsx_runtime5.jsx)("p", { children: "\xA0" }),
        /* @__PURE__ */ (0, import_jsx_runtime5.jsx)("p", { children: "Note: Internet speed may affect processing time." }),
        /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(Line, {}),
        /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(StyledParagraph, { children: "GENERATE A QR CODE:" }),
        useTestAadhaar ? /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)(import_jsx_runtime5.Fragment, { children: [
          "You can try this example app by generating a test Adhaar QR Code",
          /* @__PURE__ */ (0, import_jsx_runtime5.jsx)("p", { children: "\xA0" }),
          /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
            PhonePlatformLinks,
            {
              href: "https://documentation.anon-aadhaar.pse.dev/docs/generate-qr",
              target: "_blank",
              rel: "noreferrer",
              children: "Link to generate a QR Code"
            }
          )
        ] }) : /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)(import_jsx_runtime5.Fragment, { children: [
          "Open mAadhaar app",
          " ",
          /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
            PhonePlatformLinks,
            {
              href: "https://apps.apple.com/in/app/maadhaar/id1435469474",
              target: "_blank",
              rel: "noreferrer",
              children: "iOS"
            }
          ),
          " ",
          "or",
          " ",
          /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
            PhonePlatformLinks,
            {
              href: "https://play.google.com/store/apps/details?id=in.gov.uidai.mAadhaarPlus&hl=en_IN&pli=1",
              target: "_blank",
              rel: "noreferrer",
              children: "Android"
            }
          ),
          /* @__PURE__ */ (0, import_jsx_runtime5.jsx)("p", { children: "By entering your Aadhaar number and OTP verification. You can then save the QR as an image using the 'Share' button for import." })
        ] })
      ] }) }),
      /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(UploadSection, { children: /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)(UploadFile, { children: [
        /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)(StyledParagraph, { children: [
          "UPLOAD YOUR AADHAAR SECURE QR CODE:",
          " "
        ] }),
        /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
          FileInput,
          {
            onChange: async (e) => {
              const { qrValue } = await uploadQRpng(e, setQrStatus);
              setQrData(qrValue);
            },
            id: "handlePdfChange",
            setQrStatus,
            qrStatus
          }
        )
      ] }) })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)(Container, { children: [
      /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(Btn, { disabled: !provingEnabled, onClick: () => setCurrentView("Prove"), children: "CONTINUE" }),
      /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(SmallDisclaimer, { children: "No Aadhaar data ever leaves your device!" })
    ] })
  ] });
};
var UploadFile = import_styled_components4.default.div`
  margin-top: 20px;
  margin-bottom: 30px;
`;
var SmallDisclaimer = import_styled_components4.default.p`
  font-size: small;
  color: #717686;
  text-decoration: wavy;
  text-align: center;
  margin-top: 10px;
`;
var TitleSection = import_styled_components4.default.div`
  color: #111827;
  flex-shrink: 0;
  row-gap: 1rem;
  margin-left: auto;
  margin-right: auto;
  display: flex;
  flex-flow: column;
`;
var Disclaimer = import_styled_components4.default.span`
  color: #717686;
  font-size: 14px;
  font-weight: normal;
  line-height: 20px;
`;
var UploadSection = import_styled_components4.default.div`
  row-gap: 1rem;
  max-width: 100%;
`;
var PhonePlatformLinks = import_styled_components4.default.a`
  color: #1d24e0;
  margin-top: 0.3rem;
  font-size: 0.9rem;
  font-weight: normal;
  text-decoration: underline;
`;
var Container = import_styled_components4.default.div`
  width: 100%;
`;
var Line = import_styled_components4.default.div`
  height: 2px;
  margin: 20px 0;
  width: 5rem;
  background-color: rgba(0, 154, 8, 1);
  margin-left: auto;
  margin-right: auto;
`;
var StyledParagraph = import_styled_components4.default.p`
  font-family: 'Rajdhani', sans-serif;
  font-weight: 600; // Regular weight
  font-size: 16px; // Example font size
  color: #333; // Example text color
  margin: 10px 0;
  line-height: 1.5;
  text-transform: capitalize;
`;
var Btn = import_styled_components4.default.button`
  font-family: 'Rajdhani', sans-serif;
  font-weight: 600;
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  font-size: 16px;
  cursor: pointer;
  color: white;
  background-color: #009a08;
  border: none;
  min-width: 12rem;
  min-height: 3rem;
  border-radius: 6px;

  &:hover {
    opacity: 70%;
  }

  &:active {
    background: #f8f8f8;
  }

  &:disabled {
    color: #a8aaaf;
    background: #e8e8e8;
    cursor: default;
    cursor: not-allowed;
  }
`;
var MainContainer = import_styled_components4.default.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  width: 100%;
`;

// src/components/ProveModal/ProveModal.tsx
var import_react9 = require("react");
var import_styled_components7 = __toESM(require("styled-components"));

// src/components/ProveModal/ProveButton.tsx
var import_styled_components5 = __toESM(require("styled-components"));
var import_react8 = require("react");
var import_jsx_runtime6 = require("react/jsx-runtime");
var ProveButton = ({
  qrData,
  provingEnabled,
  setErrorMessage,
  signal,
  fieldsToReveal,
  nullifierSeed,
  setQrStatus,
  setCurrentView,
  useTestAadhaar = false
}) => {
  const { startReq } = (0, import_react8.useContext)(AnonAadhaarContext);
  const startProving = async () => {
    try {
      setCurrentView("Proving");
      if (qrData === null)
        throw new Error("Missing QR code data.");
      if (fieldsToReveal === void 0)
        fieldsToReveal = [];
      const args = await processAadhaarArgs(
        qrData,
        useTestAadhaar,
        nullifierSeed,
        fieldsToReveal,
        signal
      );
      startReq({ type: "login", args, useTestAadhaar });
      setQrStatus(null);
    } catch (error) {
      console.log(error);
      if (error instanceof Error)
        setErrorMessage(error.message);
    }
  };
  return /* @__PURE__ */ (0, import_jsx_runtime6.jsxs)(Btn2, { disabled: !provingEnabled, onClick: startProving, children: [
    " ",
    "GENERATE ANON AADHAAR PROOF",
    " "
  ] });
};
var Btn2 = import_styled_components5.default.button`
  font-family: 'Rajdhani', sans-serif;
  font-weight: 600;
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  font-size: 16px;
  cursor: pointer;
  color: white;
  background-color: #009a08;
  border: none;
  min-width: 12rem;
  min-height: 3rem;
  border-radius: 6px;

  &:hover {
    opacity: 70%;
  }

  &:active {
    background: #f8f8f8;
  }

  &:disabled {
    color: #a8aaaf;
    background: #e8e8e8;
    cursor: default;
    cursor: not-allowed;
  }
`;

// src/components/ProveModal/SignalDisplay.tsx
var import_styled_components6 = __toESM(require("styled-components"));
var import_jsx_runtime7 = require("react/jsx-runtime");
var SignalDisplay = ({
  signal
}) => {
  const isJsonObject = typeof signal === "object" && signal !== null;
  const displaySignal = isJsonObject ? JSON.stringify(signal, null, 2) : signal;
  return /* @__PURE__ */ (0, import_jsx_runtime7.jsx)(Signal, { children: displaySignal });
};
var Signal = import_styled_components6.default.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  border: solid;
  align-items: center;
  border-color: #b6b9c3;
  font-size: 14px;
  border-radius: 4px;
  color: black;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 12px;
  padding-right: 12px;
  margin-top: 10px;
`;

// src/components/ProveModal/ProveModal.tsx
var import_core5 = require("@anon-aadhaar/core");
var import_jsx_runtime8 = require("react/jsx-runtime");
var ProveModal = ({
  setErrorMessage,
  qrStatus,
  qrData,
  setQrStatus,
  signal,
  fieldsToReveal,
  nullifierSeed,
  setCurrentView,
  useTestAadhaar
}) => {
  const [provingEnabled, setProvingEnabled] = (0, import_react9.useState)(false);
  const { appName } = (0, import_react9.useContext)(AnonAadhaarContext);
  const noRevealillustration = createBlobURL(icons.eyeOff);
  const revealillustration = createBlobURL(icons.eye);
  (0, import_react9.useEffect)(() => {
    if (qrStatus === "Signature verified \u2705" /* SIGNATURE_VERIFIED */) {
      setProvingEnabled(true);
    } else {
      setProvingEnabled(false);
    }
  }, [qrStatus]);
  return /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)(MainContainer2, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)("div", { children: [
      /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(TitleSection2, { children: "YOUR QR CODE IS VERIFIED!" }),
      /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)(Section, { children: [
        /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)(Label, { children: [
          "Data you are sharing to ",
          appName,
          ": "
        ] }),
        /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(RevealSection, { children: fieldsToReveal ? import_core5.fieldsLabel.map(
          ({ key, label }) => fieldsToReveal.includes(key) ? /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(FieldRow, { children: /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)(DiscloseOn, { children: [
            /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(Icon, { src: revealillustration }),
            label.toLocaleUpperCase()
          ] }) }, key) : /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(FieldRow, { children: /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)(DiscloseOff, { children: [
            /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(Icon, { src: noRevealillustration }),
            label.toLocaleUpperCase()
          ] }) }, key)
        ) : import_core5.fieldsLabel.map(({ key, label }) => /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(FieldRow, { children: /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)(DiscloseOff, { children: [
          /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(Icon, { src: noRevealillustration }),
          label.toLocaleUpperCase()
        ] }) }, key)) })
      ] }),
      signal && /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)(Section, { children: [
        /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(Label, { children: "Data you are signing: " }),
        /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(SignalDisplay, { signal })
      ] })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)("div", { children: [
      /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(
        ProveButton,
        {
          qrData,
          provingEnabled,
          setErrorMessage,
          signal,
          setQrStatus,
          nullifierSeed,
          fieldsToReveal,
          setCurrentView,
          useTestAadhaar
        }
      ),
      /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(SmallDisclaimer2, { children: "No Aadhaar data ever leaves your device!" })
    ] })
  ] });
};
var TitleSection2 = import_styled_components7.default.div`
  font-family: 'Rajdhani', sans-serif;
  font-weight: 600; // Regular weight
  font-size: 16px; // Example font size
  color: #333; // Example text color
  line-height: 1.5;
  text-transform: capitalize;
`;
var Icon = import_styled_components7.default.img`
  height: 1.5rem;
  margin-right: 5px;
`;
var DiscloseOn = import_styled_components7.default.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  border: solid;
  align-items: center;
  border-color: #009a08;
  border-radius: 4px;
  font-family: 'Rajdhani', sans-serif;
  font-weight: 600;
  color: black;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 12px;
  padding-right: 12px;
`;
var DiscloseOff = import_styled_components7.default.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  border: solid;
  align-items: center;
  border-color: #b6b9c3;
  border-radius: 4px;
  font-family: 'Rajdhani', sans-serif;
  font-weight: 600;
  color: #b6b9c3;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 12px;
  padding-right: 12px;
`;
var Section = import_styled_components7.default.div`
  margin-top: 15px;
  row-gap: 1rem;
  max-width: 100%;
`;
var Label = import_styled_components7.default.div`
  font-size: 14px;
  text-align: left;
  font-weight: 400;
  color: #6b7280;
`;
var RevealSection = import_styled_components7.default.div`
  display: flex;
  flex-direction: column;
  row-gap: 10px;
  margin-top: 10px;
`;
var FieldRow = import_styled_components7.default.div`
  display: flex;
  align-items: center;
`;
var MainContainer2 = import_styled_components7.default.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  width: 100%;
`;
var SmallDisclaimer2 = import_styled_components7.default.p`
  font-size: small;
  color: #717686;
  text-decoration: wavy;
  text-align: center;
  margin-top: 10px;
`;

// src/components/ProveModal/LoaderView.tsx
var import_react10 = require("react");
var import_styled_components8 = __toESM(require("styled-components"));

// src/components/LoaderIcons.ts
var icons2 = {
  bottomLeftPlain: '<svg xmlns="http://www.w3.org/2000/svg" width="64" height="64" fill="none"><path fill="#081224" d="M63.813 63.116V45.86a.632.632 0 0 0-.636-.635H25.474a5.956 5.956 0 0 1-5.963-5.953V1.322a.632.632 0 0 0-.636-.635H1.385a.632.632 0 0 0-.635.635v20.545c0 2.89 1.153 5.671 3.201 7.716l30.978 30.932a10.948 10.948 0 0 0 7.706 3.196l20.534.039a.64.64 0 0 0 .644-.635"/></svg>',
  bottomLeft: '<svg xmlns="http://www.w3.org/2000/svg" width="64" height="64" fill="none"><path stroke="#081224" stroke-width=".674" d="M4.19 29.344a10.575 10.575 0 0 1-3.103-7.477V1.322c0-.166.13-.298.298-.298h17.49c.168 0 .299.132.299.298v37.95c0 3.477 2.819 6.29 6.3 6.29h37.703c.167 0 .298.132.298.298v17.255c0 .16-.133.298-.306.298l-20.533-.04a10.611 10.611 0 0 1-7.469-3.097z"/></svg>',
  bottomRightPlain: '<svg xmlns="http://www.w3.org/2000/svg" width="64" height="64" fill="none"><path fill="#FD8B0E" d="M.188 63.116V45.86c0-.353.282-.635.635-.635h37.703a5.956 5.956 0 0 0 5.963-5.953V1.322c0-.353.283-.635.636-.635h17.49c.353 0 .635.282.635.635v20.545c0 2.89-1.153 5.671-3.201 7.716L29.071 60.515a10.948 10.948 0 0 1-7.706 3.196L.831 63.75a.64.64 0 0 1-.643-.635"/></svg>',
  bottomRight: '<svg xmlns="http://www.w3.org/2000/svg" width="64" height="64" fill="none"><path stroke="#FD8B0E" stroke-width=".674" d="M59.81 29.344a10.575 10.575 0 0 0 3.103-7.477V1.322a.295.295 0 0 0-.298-.298h-17.49a.295.295 0 0 0-.299.298v37.95c0 3.477-2.819 6.29-6.3 6.29H.823a.295.295 0 0 0-.298.298v17.255c0 .16.133.298.306.298l20.534-.04a10.611 10.611 0 0 0 7.468-3.097z"/></svg>',
  topLeftPlain: '<svg xmlns="http://www.w3.org/2000/svg" width="64" height="64" fill="none"><path fill="#009A08" d="M63.813.884V18.14a.632.632 0 0 1-.636.635H25.474a5.956 5.956 0 0 0-5.963 5.953v37.95a.632.632 0 0 1-.636.635H1.385a.632.632 0 0 1-.635-.635V42.134c0-2.89 1.153-5.671 3.201-7.716L34.929 3.485A10.948 10.948 0 0 1 42.635.289L63.169.25a.64.64 0 0 1 .644.634"/></svg>',
  topLeft: '<svg xmlns="http://www.w3.org/2000/svg" width="64" height="64" fill="none"><path stroke="#009A08" stroke-width=".674" d="M4.19 34.656a10.575 10.575 0 0 0-3.103 7.477v20.545c0 .166.13.298.298.298h17.49a.295.295 0 0 0 .299-.297V24.728c0-3.477 2.819-6.29 6.3-6.29h37.703a.295.295 0 0 0 .298-.298V.884a.302.302 0 0 0-.306-.297l-20.533.04a10.611 10.611 0 0 0-7.469 3.097z"/></svg>',
  topRightPlain: '<svg xmlns="http://www.w3.org/2000/svg" width="64" height="64" fill="none"><path fill="#081224" d="M.188.884V18.14c0 .353.282.635.635.635h37.703a5.956 5.956 0 0 1 5.963 5.953v37.95c0 .353.283.635.636.635h17.49a.632.632 0 0 0 .635-.635V42.134c0-2.89-1.153-5.671-3.201-7.716L29.071 3.485A10.949 10.949 0 0 0 21.365.289L.831.25a.64.64 0 0 0-.643.634"/></svg>',
  topRight: '<svg xmlns="http://www.w3.org/2000/svg" width="64" height="64" fill="none"><path stroke="#081224" stroke-width=".674" d="M59.81 34.656a10.575 10.575 0 0 1 3.103 7.477v20.545a.295.295 0 0 1-.298.298h-17.49a.295.295 0 0 1-.299-.298v-37.95c0-3.477-2.819-6.29-6.3-6.29H.823a.295.295 0 0 1-.298-.298V.884c0-.16.133-.297.306-.297l20.534.04a10.611 10.611 0 0 1 7.468 3.097z"/></svg>'
};

// src/components/ProveModal/LoaderView.tsx
var import_core6 = require("@anon-aadhaar/core");
var import_jsx_runtime9 = require("react/jsx-runtime");
var LoaderView = () => {
  useFonts();
  const { proverState } = (0, import_react10.useContext)(AnonAadhaarContext);
  const [iconsState, setIconsState] = (0, import_react10.useState)({
    topLeft: false,
    topRight: false,
    bottomLeft: false,
    bottomRight: false
  });
  const iconsUrl = (0, import_react10.useMemo)(
    () => ({
      topLeft: createBlobURL(icons2.topLeft),
      topLeftPlain: createBlobURL(icons2.topLeftPlain),
      topRight: createBlobURL(icons2.topRight),
      topRightPlain: createBlobURL(icons2.topRightPlain),
      bottomLeft: createBlobURL(icons2.bottomLeft),
      bottomLeftPlain: createBlobURL(icons2.bottomLeftPlain),
      bottomRight: createBlobURL(icons2.bottomRight),
      bottomRightPlain: createBlobURL(icons2.bottomRightPlain)
    }),
    [icons2]
  );
  (0, import_react10.useEffect)(() => {
    const intervalId = setInterval(() => {
      setIconsState((prevState) => {
        const newState = { ...prevState };
        if (!newState.topLeft) {
          newState.topLeft = true;
        } else if (!newState.topRight) {
          newState.topRight = true;
        } else if (!newState.bottomRight) {
          newState.bottomRight = true;
        } else if (!newState.bottomLeft) {
          newState.bottomLeft = true;
        } else {
          return {
            topLeft: false,
            topRight: false,
            bottomLeft: false,
            bottomRight: false
          };
        }
        return newState;
      });
    }, 1e3);
    return () => {
      clearInterval(intervalId);
      Object.values(iconsUrl).forEach(URL.revokeObjectURL);
    };
  }, [iconsUrl]);
  return /* @__PURE__ */ (0, import_jsx_runtime9.jsxs)(import_jsx_runtime9.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(MainContainer3, { children: /* @__PURE__ */ (0, import_jsx_runtime9.jsxs)(LoaderContainer, { children: [
      /* @__PURE__ */ (0, import_jsx_runtime9.jsxs)(TopContainer, { children: [
        /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(
          Icon2,
          {
            src: iconsState.topLeft ? iconsUrl.topLeftPlain : iconsUrl.topLeft,
            alt: "Top left icon"
          }
        ),
        /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(
          Icon2,
          {
            src: iconsState.topRight ? iconsUrl.topRightPlain : iconsUrl.topRight,
            alt: "Top right icon"
          }
        )
      ] }),
      /* @__PURE__ */ (0, import_jsx_runtime9.jsxs)(BottomContainer, { children: [
        /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(
          Icon2,
          {
            src: iconsState.bottomLeft ? iconsUrl.bottomLeftPlain : iconsUrl.bottomLeft,
            alt: "Bottom left icon"
          }
        ),
        /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(
          Icon2,
          {
            src: iconsState.bottomRight ? iconsUrl.bottomRightPlain : iconsUrl.bottomRight,
            alt: "Bottom right icon"
          }
        )
      ] }),
      /* @__PURE__ */ (0, import_jsx_runtime9.jsxs)(TitleSection3, { children: [
        proverState === import_core6.ProverState.Initializing && "GENERATE PROOF...",
        proverState === import_core6.ProverState.Completed && "GENERATE PROOF...",
        proverState === import_core6.ProverState.FetchingWasm && "SEARCHING FOR WASM FILE...",
        proverState === import_core6.ProverState.FetchingZkey && "SEARCHING FOR ZKEY FILE...",
        proverState === import_core6.ProverState.Proving && "GENERATING PROOF...",
        proverState === import_core6.ProverState.Error && "Oops, something went wrong..."
      ] })
    ] }) }),
    /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(SmallDisclaimer3, { children: "Proof generation on a MacBook Pro M1 - 16GB is about 26 seconds" })
  ] });
};
var LoaderContainer = import_styled_components8.default.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 8px;
`;
var TopContainer = import_styled_components8.default.div`
  display: flex;
  flex-direction: row;
  gap: 8px;
`;
var BottomContainer = import_styled_components8.default.div`
  display: flex;
  flex-direction: row;
  gap: 8px;
`;
var Icon2 = import_styled_components8.default.img`
  height: 65px;
`;
var MainContainer3 = import_styled_components8.default.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  width: 100%;
`;
var TitleSection3 = import_styled_components8.default.div`
  margin-top: 20px;
  font-family: 'Rajdhani', sans-serif;
  font-weight: 500; // Regular weight
  font-size: 28px; // Example font size
  color: #333; // Example text color
  line-height: 1.5;
  text-transform: capitalize;
`;
var SmallDisclaimer3 = import_styled_components8.default.p`
  font-size: 12px;
  color: black;
  text-decoration: wavy;
  text-align: center;
  margin-top: 10px;
`;

// src/components/ProveModal/Modal.tsx
var import_jsx_runtime10 = require("react/jsx-runtime");
var Modal = ({
  isOpen,
  onClose,
  errorMessage,
  setErrorMessage,
  logo,
  qrStatus,
  setQrStatus,
  signal,
  fieldsToReveal,
  nullifierSeed,
  currentView,
  setCurrentView,
  useTestAadhaar = false
}) => {
  const [qrData, setQrData] = (0, import_react11.useState)(null);
  const [provingEnabled, setProvingEnabled] = (0, import_react11.useState)(false);
  (0, import_react11.useEffect)(() => {
    if (qrData) {
      verifySignature(qrData, useTestAadhaar).then((verified) => {
        verified.isSignatureValid ? setQrStatus("Signature verified \u2705" /* SIGNATURE_VERIFIED */) : setQrStatus("QR code invalid \u274C" /* ERROR_PARSING_QR */);
      }).catch((error) => {
        setQrStatus("QR code invalid \u274C" /* ERROR_PARSING_QR */);
        console.error(error);
      });
    }
  }, [qrData]);
  (0, import_react11.useEffect)(() => {
    if (qrStatus === "Signature verified \u2705" /* SIGNATURE_VERIFIED */) {
      setProvingEnabled(true);
    } else {
      setProvingEnabled(false);
    }
  }, [qrStatus]);
  return isOpen ? /* @__PURE__ */ (0, import_jsx_runtime10.jsxs)(ModalOverlay, { onClick: onClose, children: [
    /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(import_react_device_detect.BrowserView, { children: /* @__PURE__ */ (0, import_jsx_runtime10.jsxs)(ModalContent, { onClick: (e) => e.stopPropagation(), children: [
      errorMessage !== null && /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(
        ErrorToast,
        {
          message: errorMessage,
          setErrorMessage
        }
      ),
      (() => {
        switch (currentView) {
          case "Verify":
            return /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(
              VerifyModal,
              {
                provingEnabled,
                qrStatus,
                setQrStatus,
                setQrData,
                setCurrentView,
                useTestAadhaar
              }
            );
          case "Prove":
            return /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(
              ProveModal,
              {
                setErrorMessage,
                qrStatus,
                qrData,
                setQrStatus,
                signal,
                fieldsToReveal,
                nullifierSeed,
                setCurrentView,
                useTestAadhaar
              }
            );
          case "Proving":
            return /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(LoaderView, {});
        }
      })()
    ] }) }),
    /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(import_react_device_detect.MobileView, { children: /* @__PURE__ */ (0, import_jsx_runtime10.jsxs)(ModalContent, { onClick: (e) => e.stopPropagation(), children: [
      errorMessage !== null && /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(
        ErrorToast,
        {
          message: errorMessage,
          setErrorMessage
        }
      ),
      /* @__PURE__ */ (0, import_jsx_runtime10.jsxs)(TitleSection4, { children: [
        /* @__PURE__ */ (0, import_jsx_runtime10.jsxs)(Title, { children: [
          /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(Logo, { src: logo }),
          "Prove your Identity"
        ] }),
        /* @__PURE__ */ (0, import_jsx_runtime10.jsxs)(Disclaimer2, { children: [
          /* @__PURE__ */ (0, import_jsx_runtime10.jsx)("b", { children: "Notice: " }),
          " Currently, Anon Aadhaar Identity verification is not available on mobile devices. To complete this process, please visit this website using a desktop browser. We apologize for any inconvenience and thank you for your understanding."
        ] })
      ] })
    ] }) })
  ] }) : null;
};
var ModalOverlay = import_styled_components9.default.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.2);
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 9999;
`;
var ModalContent = import_styled_components9.default.div`
  position: fixed;
  display: flex;
  flex-direction: column;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: #ffffff;
  border-radius: 1rem;
  box-shadow: 0 4px 10px rgba(0, 0, 0, 0.1);
  padding: 2rem;

  @media (max-width: 425px) {
    /* For screens <= 425px (e.g., mobile devices) */
    width: 100%;
    height: 30%;
    max-width: 100%;
    max-height: 100%;
  }

  @media (min-width: 426px) {
    /* For screens > 426px (e.g., desktop) */
    min-height: 600px;
    max-width: 450px;
    width: 80%;
  }
`;
var TitleSection4 = import_styled_components9.default.div`
  color: #111827;
  flex-shrink: 0;
  row-gap: 1rem;
  margin-left: auto;
  margin-right: auto;
  display: flex;
  flex-flow: column;
`;
var Title = import_styled_components9.default.h3`
  display: flex;
  flex-shrink: 0;
  margin-left: auto;
  margin-right: auto;
  font-size: medium;
  font-weight: bold;
`;
var Disclaimer2 = import_styled_components9.default.span`
  color: #6d6d6d;
  font-size: small;
  font-weight: normal;
`;

// src/components/LogInWithAnonAadhaar.tsx
var import_styled_components10 = __toESM(require("styled-components"));
var import_react13 = require("react");
var import_core7 = require("@anon-aadhaar/core");
var import_jsx_runtime11 = require("react/jsx-runtime");
var LogInWithAnonAadhaar = ({
  signal,
  fieldsToReveal,
  nullifierSeed
}) => {
  const [isModalOpen, setIsModalOpen] = (0, import_react12.useState)(false);
  const [errorMessage, setErrorMessage] = (0, import_react12.useState)(null);
  const [qrStatus, setQrStatus] = (0, import_react12.useState)(null);
  const [currentView, setCurrentView] = (0, import_react12.useState)("Verify");
  const [isMenuOpen, setIsMenuOpen] = (0, import_react12.useState)(false);
  const { state, startReq, proverState, useTestAadhaar } = (0, import_react13.useContext)(AnonAadhaarContext);
  const anonAadhaarLogo = createBlobURL(icons.aalogo);
  const toggleMenu = () => {
    setIsMenuOpen(!isMenuOpen);
  };
  (0, import_react13.useEffect)(() => {
    if (proverState === import_core7.ProverState.Completed)
      closeModal();
  }, [proverState]);
  const openModal = () => {
    setIsModalOpen(true);
  };
  const closeModal = () => {
    setIsModalOpen(false);
    setErrorMessage(null);
    setQrStatus(null);
    setCurrentView("Verify");
  };
  return /* @__PURE__ */ (0, import_jsx_runtime11.jsxs)("div", { children: [
    (state.status === "logged-out" || state.status === "logging-in") && /* @__PURE__ */ (0, import_jsx_runtime11.jsxs)("div", { children: [
      /* @__PURE__ */ (0, import_jsx_runtime11.jsxs)(Btn3, { onClick: openModal, children: [
        /* @__PURE__ */ (0, import_jsx_runtime11.jsx)(Logo, { src: anonAadhaarLogo }),
        "Login"
      ] }),
      /* @__PURE__ */ (0, import_jsx_runtime11.jsx)(
        Modal,
        {
          isOpen: isModalOpen,
          onClose: closeModal,
          errorMessage,
          setErrorMessage,
          logo: anonAadhaarLogo,
          qrStatus,
          setQrStatus,
          signal,
          fieldsToReveal,
          nullifierSeed,
          setCurrentView,
          currentView,
          useTestAadhaar
        }
      )
    ] }),
    state.status === "logged-in" && /* @__PURE__ */ (0, import_jsx_runtime11.jsxs)(RelativeContainer, { children: [
      /* @__PURE__ */ (0, import_jsx_runtime11.jsxs)(Btn3, { onClick: toggleMenu, children: [
        /* @__PURE__ */ (0, import_jsx_runtime11.jsx)(Logo, { src: anonAadhaarLogo }),
        "Menu"
      ] }),
      /* @__PURE__ */ (0, import_jsx_runtime11.jsxs)(MenuContainer, { $isopen: isMenuOpen, children: [
        /* @__PURE__ */ (0, import_jsx_runtime11.jsx)(MenuItem, { onClick: openModal, children: "Create a proof" }),
        /* @__PURE__ */ (0, import_jsx_runtime11.jsx)(
          Modal,
          {
            isOpen: isModalOpen,
            onClose: closeModal,
            errorMessage,
            setErrorMessage,
            logo: anonAadhaarLogo,
            qrStatus,
            setQrStatus,
            signal,
            fieldsToReveal,
            nullifierSeed,
            setCurrentView,
            currentView,
            useTestAadhaar
          }
        ),
        /* @__PURE__ */ (0, import_jsx_runtime11.jsx)(MenuItem, { onClick: () => startReq({ type: "logout" }), children: "Logout" })
      ] })
    ] })
  ] });
};
var Logo = import_styled_components10.default.img`
  height: 1.5rem;
  margin-right: 0.5rem;
`;
var Btn3 = import_styled_components10.default.button`
  display: flex;
  padding: 0 1rem;
  font-size: 1rem;
  cursor: pointer;
  color: #000000;
  font-weight: bold;
  border-radius: 1.3125rem;
  background: #fff;
  box-shadow: 0px 3px 8px 1px rgba(0, 0, 0, 0.25);
  border: none;
  min-height: 2.5rem;
  border-radius: 0.5rem;
  align-items: center;

  &:hover {
    background: #fafafa;
  }

  &:active {
    background: #f8f8f8;
  }

  &:disabled {
    color: #a8aaaf;
    background: #e8e8e8;
    cursor: default;
  }
`;
var MenuItem = import_styled_components10.default.button`
  display: block;
  width: 100%;
  padding: 0.5rem 1rem;
  font-size: 1rem;
  color: #000000;
  text-align: left;
  background: none;
  border: none;
  border-bottom: 1px solid #cccccc;
  cursor: pointer;

  &:hover {
    border-radius: 0.5rem;
    background-color: #f2f2f2;
  }

  &:last-child {
    border-bottom: none;
  }
`;
var MenuContainer = import_styled_components10.default.div`
  display: ${(props) => props.$isopen ? "block" : "none"};
  position: absolute;
  margin-top: 0.5rem;
  top: 100%;
  right: 0;
  width: 130%;
  background: #fff;
  box-shadow: 0 8px 16px rgba(0, 0, 0, 0.2);
  border-radius: 0.5rem;
  z-index: 10;
`;
var RelativeContainer = import_styled_components10.default.div`
  position: relative;
  display: inline-block;
`;

// src/components/LaunchProveModal.tsx
var import_react14 = require("react");
var import_styled_components11 = __toESM(require("styled-components"));
var import_react15 = require("react");
var import_core8 = require("@anon-aadhaar/core");
var import_jsx_runtime12 = require("react/jsx-runtime");
var LaunchProveModal = ({
  signal,
  buttonStyle,
  fieldsToReveal,
  nullifierSeed,
  buttonTitle = "Generate a proof"
}) => {
  const [isModalOpen, setIsModalOpen] = (0, import_react14.useState)(false);
  const [errorMessage, setErrorMessage] = (0, import_react14.useState)(null);
  const [qrStatus, setQrStatus] = (0, import_react14.useState)(null);
  const [currentView, setCurrentView] = (0, import_react14.useState)("Verify");
  const { proverState, useTestAadhaar } = (0, import_react15.useContext)(AnonAadhaarContext);
  const anonAadhaarLogo = createBlobURL(icons.aalogo);
  (0, import_react15.useEffect)(() => {
    if (proverState === import_core8.ProverState.Completed)
      closeModal();
  }, [proverState]);
  const openModal = () => {
    setIsModalOpen(true);
  };
  const closeModal = () => {
    setIsModalOpen(false);
    setErrorMessage(null);
    setQrStatus(null);
    setCurrentView("Verify");
  };
  return /* @__PURE__ */ (0, import_jsx_runtime12.jsxs)("div", { children: [
    /* @__PURE__ */ (0, import_jsx_runtime12.jsxs)(Btn4, { style: buttonStyle, onClick: openModal, children: [
      /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(Logo2, { src: anonAadhaarLogo }),
      buttonTitle
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(
      Modal,
      {
        isOpen: isModalOpen,
        onClose: closeModal,
        errorMessage,
        setErrorMessage,
        logo: anonAadhaarLogo,
        qrStatus,
        setQrStatus,
        signal,
        fieldsToReveal,
        nullifierSeed,
        setCurrentView,
        currentView,
        useTestAadhaar
      }
    )
  ] });
};
var Logo2 = import_styled_components11.default.img`
  height: 1.5rem;
  margin-right: 0.5rem;
`;
var Btn4 = import_styled_components11.default.button`
  display: flex;
  padding: 0 1rem;
  font-size: 1rem;
  cursor: pointer;
  color: #000000;
  font-weight: bold;
  border-radius: 1.3125rem;
  background: #fff;
  box-shadow: 0px 3px 8px 1px rgba(0, 0, 0, 0.25);
  border: none;
  min-height: 2.5rem;
  border-radius: 0.5rem;
  align-items: center;

  &:hover {
    background: #fafafa;
  }

  &:active {
    background: #f8f8f8;
  }

  &:disabled {
    color: #a8aaaf;
    background: #e8e8e8;
    cursor: default;
  }
`;

// src/hooks/useProver.ts
var import_react16 = require("react");
var import_core9 = require("@anon-aadhaar/core");
function useProver() {
  const [latestProof, setLatestProof] = (0, import_react16.useState)();
  const { proverState, state } = (0, import_react16.useContext)(AnonAadhaarContext);
  (0, import_react16.useEffect)(() => {
    if (state.status === "logged-in") {
      (0, import_core9.deserialize)(
        state.anonAadhaarProofs[Object.keys(state.anonAadhaarProofs).length - 1].pcd
      ).then((anonAadhaarCore) => setLatestProof(anonAadhaarCore));
    }
  }, [proverState, state]);
  return [proverState, latestProof];
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AnonAadhaarProof,
  AnonAadhaarProvider,
  LaunchProveModal,
  LogInWithAnonAadhaar,
  processAadhaarArgs,
  proveAndSerialize,
  useAnonAadhaar,
  useProver,
  verifySignature
});
